# Methods

## DNA extraction



 

##Library preparation for targeted enrichment



Library preparation and hybridisation was carried out at the Sackler Phylogenomic Laboratory, within the Jodrell Laboratory at Royal Botanic Gardens, Kew (Richmond, Surrey, UK). The DNA concentration present in the extracts was measured with a QuantusTM fluorometer (Promega corporation, Madison USA) using the using QuantiFluorR dsDNA System kit (Promega corporation, Madison USA). The samples were diluted to 200 ng DNA in 26 $\mu$l with Milli-Q water. The DNA was then sheared to fragments an average of 350 bp long by sonicating them for 75s with a Covaris Focused-ultrasonicator M220 (Woburn Massachusetts 01801 www.covarisinc.com)



Library preparation was completed using NEBNxet Ultra II DNA Library Prep Kit for Illumina (New England Biolabs), using half the recommended volumes to maximise use of reagents and indexes. AMPure magnetic beads were used for size selection of DNA fragments. Libraries were indexed with the NEBNext Multiplex Oligos for Illumina (Dual Index Primer Set 1, New England Biolabs), and enriched with the following thermocycler conditions: Initial denaturation for 30s at 98$^\circ$C, 7 cycles of denaturation for 10s at 98$^\circ$C, annealing and extension for 75s at 65$^\circ$C, followed by a final extension for 5 mins at 65$^\circ$C. The resulting enriched libraries were cleaned with AMPure beads, and the quality of the libraries were checked using Agilent Technologies 4200 Tapestation (Agilent Technologies, Palo Alto, California, USA). A library was considered good quality if it had a concentration of greater than 4 ng.$\mu$l^-1^ and avg fragment size of 450 bp to 500 bp. If a library failed the quality check, it was either re-amplified, underwent size selection again, or was entirely repeated from the beginning of the process. 

 

## Hybridisation and sequencing preparation



Libraries were normalised to a concentration of 10 nM and 10 $\mu$l per sample was pooled into batches of 12 or 24 samples per reaction. The pools were vacuum dried using a miVac Duo Concentrator (Genevac, UK) and subsequently eluted in 8 $\mu$l of 10 mM tris buffer. The pools were enriched using the Angiosperms-353 v 1 target capture kit (Johnson et al., 2018) following the manufacturers instructions (Arbor Biosciences, Ann Arbor, Michigan, USA). Hybridisation took place for 16h at 65$^\circ$C, and the resulting product was amplified with 12 cycles of PCR. The reaction quality was checked using a Agilent Technologies 4200 Tapestation (Agilent Technologies, Palo Alto, California, USA). Reactions were normalised to 4mM and pooled. Sequencing took place either at Macrogen (Seoul, South Korea) or on an Illumina® MiSeq at Royal Botanic Gardens, Kew (Richmond, Surrey, UK). Both produced 2 x 150 bp reads.

 

## Phylogenetic methods



**Read trimming and quality control**

The raw reads were cleaned using trimmomatic V0.39 (ref). First, Illuminaclip palindrome mode using the reference file TruSeq3-PE.fa removed Illumina sequencing adaptors. Palindrome mode had the following settings: SeedMismatches = 1, palindromeClipThreshold = 30, simpleClipThreshold = 6, minAdapterLength  =  2; keepBothReads = T RUE.  Second, poor quality reads were trimmed with the Maxinfo option. Maxinfo balances the utility of retaining reads of an informative length against the costs of incorrect bases. Initially, Maxinfo is lenient towards potential error calls at a shorter read length, and becomes more strict as read length increases. The settings used were targetLength = 40 and strictness = 85. Finally, reads shorter than 36 bases were removed using MinLength. Overall read quality and adaptor content of the remaining reads were checked using FastQC 0.11.8 (ref)



**Read assembly**

Reads were assembled using the pipeline HybPiper 1.3.1 (ref).  HybPiper consists of a series of python scripts that use SPAdes (ref) and other bioinfomatic tools to assemble reads to a target file. The pipeline first sorts and groups together individual reads for a gene by mapping each read to the gene sequence in the target file.  Once the reads are sorted by gene, they are de novo assembled using SPAdes, and then realigned to the reference sequence. If multiple contigs are assembled per gene, the contig with either the greatest depth or greatest similarity to the reference file is chosen for the consensus sequence. Hybpiper gives the option of two different mapping algorithms, BLASTX and bwa. I ran the pipeline using BLASTX, as it is a less strict algorithm than bwa, therefore maximising the number of reads mapped and the lengths of the final sequences. A coverage of four reads was considered sufficient to create a consensus sequence. Reads were mapped to a multi-species, amino acid target file matching the probes from the Angiosperm 353 baitset. In the case of a multi-species target file, such as the one used here, HybPiper can determine which target sequence for a gene is the most suitable given the sample.

The Hybpiper pipeline was also used to assemble the cleaned reads of a full genome of _Oryza sativa_ Japonica cultivar downloaded from Genbank (https://www.ncbi.nlm.nih.gov/sra/SRR10389404).



**Identification of paralogs, gene choice**

Paralogs are genes related by an ancestral duplication event, rather than by sharing a common origin as is the case with orthologous genes (Koonin 2005, Schrempf2020). A key assumption of phylogenetics is that the genes used within an analysis are orthologous. Fortunately, HypPiper is able to identify potential paralogs. If HybPiper 1.3.1 recovers two or more similar contigs that cover at least 85% of the coding region's reference sequence, it flags the region as containing potential paralogs. Note that two similar length contigs for a gene may also indicate the presence of an allele. The two can be differentiated by constructing a gene tree. A gene is paralogous if the main contigs cluster separately to the alternative contigs, as this indicates a gene duplication event. Alternatively, main and secondary contigs grouping as sisters within a gene would suggest alleles.

In this analysis, HybPiper flagged 50 genes with paralog warnings. Within each gene, the number of samples flagged with paralogs varied between 1 and 39 samples. I disregarded genes with less than 2 flagged samples, as these were more likely to be alleles rather than paralogs. This left 18 genes that contained potential paralogs. HybPiper contains a script, paralog_retriever.py that will extract both the main and secondary contigs for a gene into a fasta file. Each resulting fasta was then aligned with MAFFT, followed by genetree construction using IQtree 1.6.11.  The trees were inspected in Geneious. While it wasn't always clear cut whether the tree exhibited paralogs or not, I errored on the side of caution and removed any gene that wasn't obviously allelic. Therefore a total of 11 genes were deemed paralogs, and were removed from the dataset.

Two sets of loci were chosen for further phylogenetic analysis. One, refered to as the full coverage dataset (FCD), retained genes that were present in all 109 samples, outgroups and rice included. This was a total of 235 genes (Table). The other, referred to as the ninety percent coverage dataset (NCD), retained genes that were present in at least 90% of all samples. This cut off was chosen as a compromise between missing data, and using as many genes as possible. These dataset divisions were used for exons, supercontigs, and introns, giving six datasets in total. Individual genes were aligned using MAFFT 7.427 with the settings --localpair --maxiterate 1000. The resulting alignments were then trimmed of gappy and poorly aligned regions with TrimAL 1.2. 



**Maximum likelihood trees** 

The alignments were concatenated using AMAS (ref), which also generates a partition file delimiting where each loci starts in the concatenated alignment. Most loci have a unique evolutionary history, and are able to evolve independently of the rest of the genome. Each region is therefore expected to have a different rate and pattern of evolution. Therefore each locus should be fitted with a model of evolution that best fits its history. To reduce parameter space, it is also best to partition together loci with similar models of evolution. IQtree is able to fit 200 odd models of evolution to the genes and can implement Lanfear's partitionfinder2 (Chernomor 2016, Lanfear). Due to the high computational burden, I choose to use the relaxed hierarchical clustering algorithm (-rcluster) (Lanfear2014), which only examines the top n% partition merging schemes. The exon supermatrices were partitioned using -rcluster 10, while the intron and supercontig supermatrices were run with -rcluster 2 due to computational time limits. The locus limits were used as the initial partitions.

I then reconstructed ML trees on the merged partition datasets using IQtree v. For support values, I used 1000 ultrafast bootstraps (UFBoot, Hoang 2017), and Shimodaira-Hasegawa approximate likelihood-ratio test (SH-aLRT) (Shimodaira 1999, Guindon 2010) implemented with -alrt 1000. SH-aLRT calculates if there is a significant difference between the likelihoods of the best and the second best topological arrangements around a given branch (Anisimova 2006, Guindon 2010). With these support metrics, a node is considered to have good support if it has greater than 95% UFBoot value, and greater than 80% Sh-aLRT value.



**Multi-species coalescent trees**



The evolutionary history of a gene often differs from the history of a species due to incomplete lineage sorting, horizontal gene transfer and hybridisation resulting in gene trees that differ from the true species tree. Concatenated ML methods are unlikely to capture these patterns. Conversely, the multi-species coalescent model allows the history genes to be modeled separately from the species history, therefore allowing for processes such as ILS to occur. Summary-based methods of species tree estimation such as ASTRAL are consistent with multi-species coalescent theory, and are therefore thought to be more accurate than ML for large datasets. Astral estimates the species tree that has the most quartets that are shared with the gene trees. Individual gene trees were constructed from the loci alignments using IQTree with the default settings. ASTRAL-III v5.6.3 was used to infer the species tree from the individual gene trees. As coalescent-based methods are less sensitive to missing taxa (Hosner 2016, Sayyari 2017), I used only the ninety-percent coverage datasets for this analysis.





