# Methods

## Library preparation

**Quantus** (QuantusTM fluorometer, Promega corporation, Madison USA) using (QuantiFluorR dsDNA System, Promega corporation, Madison USA)

\-    Dye = 1990 ul 1X TE buffer 10ul dye

\-    Blank = 100ul 1X TE buffer, 100 ul dye

\-    Std = 98 ul 1X TE buffer, 100 ul dye, 2 ul dna standard

\-    Samples = 99 ul TE buffer, 100 ul dye, 1 ul sample

Gives reading of DNA concentration

 

**Dilution**

Need 200 ng DNA in 26 ul liquid. Liquid used was MilliQ water to dilute our TE buffer elute a bit.

Calculations: Vol sample used = 200/quantus concentration. MilliQ water added = 26 ul – vol sample used.

 

**Sonication** (Covaris Focused-ultrasonicator M220, software Covaris SonoLabTM 2013 Version 7.2.1.0, Woburn Massachusetts 01801 www.covarisinc.com)

Transfer diluted sample into covaris tubes

Sonicated for 75 secs @ temp of between 18 and 20C

Avg Incident power 10 Watts, Peak Incident power 50W, Duty factor 20%, cycles/burst 200)

Aiming for 350 bp

 

**LIBRARY PREPARATION**

NEBNxet Ultra II DNA Library Prep Kit for Illumina. New England Biolabs inc.

Note: We were using half the recommended reagent quantities with the same amount of DNA (200ng)

www.neb.com

 

**End Prep** repairing torn ends after sonication

Kit enzymes and buffer added. In thermocycler for 30 mins @ 20C and 30 mins @ 65C

 

**Adapter ligation** adds adapters to the repaired ends

Add master mix and ligation enhancer

In thermocycler for 15 mins @ 20C

 

**USER,** in previous step adapters attach to the ends of two different strands, USER enzymes cuts the adapters in half 

In thermocycler for 15 mins @ 37C

 

**Size selection of DNA fragments**

Add 13 ul AMPure magnetic beads to 95 ul of the library. Should bind to long fragments. Discard the beads with the long fragments and keep supernatant. Add 5ul beads to supernatant. All dna except shortest fragments should bind to the beads. Discard supernatant. Wash and re elute beads in 10mM tris.

 

**PCR enrichment**

 

Clean pipettes and bench with bleach 

Add master mix from kit and two different index primers – i7 and i5

NEBNext Multiplex Oligos for Illumina (Dual Index Primer Set 1) #E7600S New England Biolabs Inc. 

 

Thermocycler conditions

 

Initial denaturation - 30s @ 98C

 

Denaturation – 10s @ 98C

Annealing/Extension – 75s @ 65C

Repeat above for 7 or 8 cycles 

 

Final extension – 5 mins @ 65C

 

**Clean up of PCR amplification**

 

Add 22.5 ul Ampure magnetic beads to sample.

Discard supernatant and wash beads with 80% ethanol

Elute beads in 16.5 ul 10mM tris.

Transfer 15.5 ul supernantant to new well.

 

**Library quality check**

 

Quantus as above for library concentration

 

Check fragment size distribution using Agilent Technologies 4200 Tapestation

And Agilent Technologies 4200 Tapestation Controller software A.02.02 (SR1)

\+ get average fragment size using the region tool in Agilent Technologies 4200 Tapestation Analysis software A.02.02 (SR1)

Using DS1000 tapes

3 ul buffer from kit + 1 ul sample. Vortex and spin down. Ladder from kit.

 

Aiming for a library concentration of 4 ng/ul and avg fragment size of 450.

In reality mine were much more concentrated with a frag size of ~500

 

If libraries failed, either rePCR, re size selected, or redo from the beginning.

rePCR – primers Universal primers P05 and P07 from bait kit.

 

 

**SEQUENCING PREPARATION**

 

**Normalising samples**

 

All libraries need to be at a concentration of 10 nM.

Calculate current molarity: (Lib. Conc)/(Avg. frag size * 660) *1000000

(The 660 is the average molecular weight of a base pair)

 

Add 5 ul of sample to a well.

Calculate the amount of water needed to dilute to 10 nM with (lib. Molarity – 10)/2

 

**Pooling samples**

 

I didn’t have to worry about pooling by relatedness or quality as all mine were pretty similar.

 

Add 10 ul of 24 samples (12 samples in the case of the MiSeq) to a 1.5 ml epi. Vac dry for 30-40 mins in miVac DUO concentrator (genevac)

 

Elute in 8 ul of 10 mM tris

 

**HYBRIDISATION**

 

Kit used:

Arbor Biosciences myBaits® Target Capture Kit Box 1 Ref#190206-30 and Box 2

Arborbiosci.com

Hybridisation capture for targeted NGS Manual v.4.01

 

**Reaction assembly**

 

Mix together Hyb reagents and baits

Mix together blocks

 

Add libraries to block mixture

Incubate at 95C for 5 mins

 

Bring hybridization mixture to 95C

Add to libraries/block mixture

 

Incubate at 65C for 16hours

 

**Clean up**

 

Prepare wash buffer and heat to 65C

 

Wash beads from kit (not Ampure) with binding buffer 3 times and resuspend in binding buffer, warm to 65C

 

Add libraries/block mixture to beads and incubate for 5 mins @ 65C

 

Pellet, wash and resuspend beads three times. Discard supernatant

 

**Resuspension and amplification**

 

Elute beads in 30 ul of Tris and Tween-20 solution 

Add Universal primers P05 and P07 and blue mastermix from library kit.

 

Thermocycler program:

 

98C for 2 mins

 

98C for 20s

55C for 30s

72C for 45s 

Repeat above three for 12 cycles

 

72C for 5 mins

 

 

**Prep for MiSeq**

 

Normalise to 4mM (My library was pretty much there anyway at 3.8 mM, so skipped this step)

 

Dilute 2M NaOH to 0.2M NaOH

 

Add 5ul library to 5ul NaOH (denaturing)

 

Add 990 Hyb buffer from illumina kit

 

Take 330 ul of that solution and dilute it with 270 ul more buffer to make a 11 pM solution.

 

Load into MiSeq



## Phylogenetic methods



**Read trimming and quality control**

The raw reads were cleaned using trimmomatic V0.39 (ref). First, Illuminaclip palindrome mode using the reference file TruSeq3-PE.fa removed Illumina sequencing adaptors. Palindrome mode had the following settings: SeedMismatches = 1, palindromeClipThreshold = 30, simpleClipThreshold = 6, minAdapterLength  =  2; keepBothReads = T RUE.  Second, poor quality reads were trimmed with the Maxinfo option. Maxinfo balances the utility of retaining reads of an informative length against the costs of incorrect bases. Initially, Maxinfo is lenient towards potential error calls at a shorter read length, and becomes more strict as read length increases. The settings used were targetLength = 40 and strictness = 85. Finally, reads shorter than 36 bases were removed using MinLength. Overall read quality and adaptor content of the remaining reads were checked using FastQC 0.11.8 (ref)



**Read assembly**

Reads were assembled using the pipeline HybPiper 1.3.1 (ref).  HybPiper consists of a series of python scripts that use SPAdes (ref) and other bioinfomatic tools to assemble reads to a target file. The pipeline first sorts and groups together individual reads for a gene by mapping each read to the gene sequence in the target file.  Once the reads are sorted by gene, they are de novo assembled using SPAdes, and then realigned to the reference sequence. If multiple contigs are assembled per gene, the contig with either the greatest depth or greatest similarity to the reference file is chosen for the consensus sequence. Hybpiper gives the option of two different mapping algorithms, BLASTX and bwa. I ran the pipeline using BLASTX, as it is a less strict algorithm than bwa, therefore maximising the number of reads mapped and the lengths of the final sequences. A coverage of four reads was considered sufficient to create a consensus sequence. Reads were mapped to a multi-species, amino acid target file matching the probes from the Angiosperm 353 baitset. In the case of a multi-species target file, such as the one used here, HybPiper can determine which target sequence for a gene is the most suitable given the sample.

The Hybpiper pipeline was also used to assemble the cleaned reads of a full genome of _Oryza sativa_ Japonica cultivar downloaded from Genbank (https://www.ncbi.nlm.nih.gov/sra/SRR10389404).



**Chloroplast**



**Identification of paralogs, gene choice**

Paralogs are genes related by an ancestral duplication event, rather than by sharing a common origin as is the case with orthologous genes (Koonin 2005, Schrempf2020). A key assumption of phylogenetics is that the genes used within an analysis are orthologous. Fortunately, HypPiper is able to identify potential paralogs. If HybPiper 1.3.1 recovers two or more similar contigs that cover at least 85% of the coding region's reference sequence, it flags the region as containing potential paralogs. Note that two similar length contigs for a gene may also indicate the presence of an allele. The two can be differentiated by constructing a gene tree. A gene is paralogous if the main contigs cluster separately to the alternative contigs, as this indicates a gene duplication event. Alternatively, main and secondary contigs grouping as sisters within a gene would suggest alleles.

In this analysis, HybPiper flagged 50 genes with paralog warnings. Within each gene, the number of samples flagged with paralogs varied between 1 and 39 samples. I disregarded genes with less than 2 flagged samples, as these were more likely to be alleles rather than paralogs. This left 18 genes that contained potential paralogs. HybPiper contains a script, paralog_retriever.py that will extract both the main and secondary contigs for a gene into a fasta file. Each resulting fasta was then aligned with MAFFT, followed by genetree construction using IQtree 1.6.11.  The trees were inspected in Geneious. While it wasn't always clear cut whether the tree exhibited paralogs or not, I errored on the side of caution and removed any gene that wasn't obviously allelic. Therefore a total of 11 genes were deemed paralogs, and were removed from the dataset.

Two sets of loci were chosen for further phylogenetic analysis. One, refered to as the full coverage dataset (FCD), retained genes that were present in all 109 samples, outgroups and rice included. This was a total of 235 genes (Table). The other, referred to as the ninety percent coverage dataset (NCD), retained genes that were present in at least 90% of all samples. This cut off was chosen as a compromise between missing data, and using as many genes as possible. These dataset divisions were used for exons, supercontigs, and introns, giving six datasets in total. Individual genes were aligned using MAFFT 7.427 with the settings --localpair --maxiterate 1000. The resulting alignments were then trimmed of gappy and poorly aligned regions with TrimAL 1.2. 



**Maximum likelihood trees** 

The alignments were concatenated using AMAS (ref), which also generates a partition file delimiting where each loci starts in the concatenated alignment. Most loci have a unique evolutionary history, and are able to evolve independently of the rest of the genome. Each region is therefore expected to have a different rate and pattern of evolution. Therefore each locus should be fitted with a model of evolution that best fits its history. To reduce parameter space, it is also best to partition together loci with similar models of evolution. IQtree is able to fit 200 odd models of evolution to the genes and can implement Lanfear's partitionfinder2 (Chernomor 2016, Lanfear). Due to the high computational burden, I choose to use the relaxed hierarchical clustering algorithm (-rcluster) (Lanfear2014), which only examines the top n% partition merging schemes. The exon supermatrices were partitioned using -rcluster 10, while the intron and supercontig supermatrices were run with -rcluster 2 due to computational time limits. The locus limits were used as the initial partitions.

I then reconstructed ML trees on the merged partition datasets using IQtree v. For support values, I used 1000 ultrafast bootstraps (UFBoot, Hoang 2017), and Shimodaira-Hasegawa approximate likelihood-ratio test (SH-aLRT) (Shimodaira 1999, Guindon 2010) implemented with -alrt 1000. SH-aLRT calculates if there is a significant difference between the likelihoods of the best and the second best topological arrangements around a given branch (Anisimova 2006, Guindon 2010). With these support metrics, a node is considered to have good support if it has greater than 95% UFBoot value, and greater than 80% Sh-aLRT value.



**Multi-species coalescent trees**





Individual gene trees were constructed from the loci alignments using IQTree with the default settings. ASTRAL-III v5.6.3. As coalescent-based methods are less sensitive to missing taxa (Hosner 2016, Sayyari 2017), I used only the ninety-percent coverage datasets for this analysis.





