---
title: "Phylogenetic results"
author: "Lara Wootton"
date: "11/05/2020"
output: 
  pdf_document
---

```{r setup, include=F}
knitr::opts_chunk$set(cache = T,
                      echo=FALSE, warning=FALSE, message=FALSE)

if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr, tidyverse, here)


```



## Sequencing results



#### Targeted-enrichment dataset

I sequenced 353 loci for 108 taxa, with the addition of _O. sativa_ as an outgroup. After trimming, an average of 228907 reads per sample were mapped to target loci. Of the 353 loci, an average of 200 loci per sample had reads mapped to at least 75% of their length. Phylogenies generated from the ninety-coverage, full-coverage, exon, intron and supercontig datasets contained similar topologies. Therefore, only results obtained from the ninety-coverage supercontig dataset are presented. The ninety-coverage supercontig dataset contained 235 loci, each represented in a minimum of 98 taxa. The concatenated alignment used for maximum-likelihood analyses was 698832 bp long, and contained 12.5% gap characters. There were 388704 (55.6%) variable sites, and 226897 (32.5%) parsimony-informative sites. The percent GC content was 39.4%. Partitionfinder reduced the number of partitions from 235 to 84.



#### Genotyping-by-sequencing dataset







## Phylogenetic results

The maximum-likelihood (ML) and maximum quartet support (MQS) methods yielded well-supported, topologically-similar trees (Figure \@ref(fig:MLtree), Figure \@ref(fig:astraltree)). In both trees, six major clades (here termed the Australian, Rehmannii, Ramosa, Rupestris, Dura and Lowland clades) are resolved with strong support, the relationships of these clades also being consistently well supported. The branches subtending these major clades are long, with branch length in the ML tree reflecting sequence divergence and that in the MQS tree numbers of coalescent units (generations/effective population size). In the Rupestris, Ramosa and Rehmannii clades, internal branches tend to be very short relative to both the stem and terminal branches, resulting in a “broomstick” morphology (ref) (Figure \@ref(fig:MLtree)). In contrast, the Lowland clade has generally longer branches, with branch lengths being more evenly distributed throughout the clade. The Australian clade, comprising *M. stipoides* and *T. laevis* (BS = 100, LPP = 1.00), is consistently resolved as sister to the predominantly-montane Rupestris clade (BS = 100, LPP = 1.00) with strong support (BS = 100, LPP = 1.00), this pair being sister to the rest of *Ehrharta*. The Rupestris clade is composed of two species, *E. setacea* and *E. rupestris*, neither of which is monophyletic. 

The montane Dura clade (BS = 100, LPP = 1.00), comprising *E. dura* and *E. microlaena*, is resolved with strong support (BS = 100, LPP = 1.00) as sister to a well-supported clade (BS = 100, LPP = 1.00) comprising the Rehmannii, Ramosa and Lowland clades. Within this clade, the Rehmannii (BS = 100, LPP = 1.00) and Ramosa clades (BS = 100, LPP = 1.00), each comprising a single species, are resolved as sisters with strong support (BS = 100, LPP = 1.00). The monophyly of the Lowland clade, which comprises species occurring predominately at low to mid elevations in the GCFR, is also well supported (BS = 100, LPP = 1.00), with the two analyses resolving the internal relationships of this clade identically and with maximum support (BS = 100, LPP = 1.00) on most nodes. Relationships within the Lowlands clade are largely consistent with those reported by Verboom et al. (2003).





```{r MLtree, echo=F, fig.height=10, fig.width=9, warning=F, fig.cap="Maximum-likelihood tree constructed using IQtree. Red numbers show ultrafast bootstraps. Branch lengths represent number of substitutions. The outgroup branch length, coloured grey, has been shortened by a factor of 2.5 for clarity"}
#ML tree

tree <- read.tree(here("Data","boot_relaxed_clustering2_ninety_coverage_supercontig_21Apr.treefile"))
tree <- root(tree, outgroup = "Oryza_sativa", resolve.root = T) #Re-root

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree$tip.label[tree$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Species[[i]],
        paste("(",
        species_names$SampleID[[i]],
        ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

tree$tip.label[tree$tip.label == "Tetrarrhena laevis () "] <- "Tetrarrhena laevis"
tree$tip.label[tree$tip.label == "Microlaena stipoides () "] <- "Microlaena stipoides"

#For visualisation purposes, cut the outgroup branch length by half
tree$edge.length[length(tree$edge.length)] <- tree$edge.length[length(tree$edge.length)]/2

gg_tree <- ggtree(tree, right = T)
d <- gg_tree$data
d <- d[!d$isTip,]
d <- d[!d$label=="Root",]
d <- separate(d, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)
da <- d[d$node %in% c(167, 119, 217, 118,117, 120,168,170,189,171,113),]

low <- gg_tree$data
lowl <- low[low$node %in% c(111,203,204,205,206,207,208,
                            209,210,211,212,213,214,215,
                            216,114,115,116,112),]
lowl <- separate(lowl, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)
lowl$bootstrap[lowl$node == 111] <- "100"

tree_group <- groupClade(tree, .node = 217)

ggtree(tree_group, aes(colour = group), right = T) +
    scale_colour_manual(values = c("grey", "black")) +
    geom_tiplab(size = 2, colour = "black") +
    geom_treescale(x = -0.001, offset = 0.6, linesize = 0.6) +
    geom_nodelab(data = da, aes(label = bootstrap),
                 size = 2.5,
                 colour = "red",
                 nudge_x = -0.0033,
                 nudge_y = -0.69) +
    geom_nodelab(data = lowl, aes(label = bootstrap),
                 size = 2.5,
                 colour = "red",
                 nudge_x = -0.0033,
                 nudge_y = -0.69)+
    geom_rootedge(rootedge = 0.01)+
    
    #Major clades
    
    geom_cladelabel(node = 167, 
                    label = "Australian clade",
                    offset = 0.022,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.5) +
    geom_cladelabel(node = 113, 
                    label = "Lowlands clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.2,
                    offset = 0.035,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001) +
    geom_cladelabel(node = 189, 
                    label = "Rehmannii clade",
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.2,
                    offset = 0.055,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001) +
    geom_cladelabel(node = 171, 
                    label = "Ramosa clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.2,
                    offset = 0.062,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001
    ) +
    geom_cladelabel(node = 120, 
                    label = "Rupestris clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.3,
                    offset = 0.07,
                    #angle = 90,
                   # hjust = 0.5,
                    offset.text = 0.001) +
    geom_cladelabel(node = 168, 
                    label = "Dura clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.3,
                    offset = 0.025,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001) +
    geom_cladelabel(node = 116, 
                    label = "Dummy clade", 
                    #align = T,
                    colour = "white",
                    fontsize = 3,
                    extend = 0.3,
                    offset = 0.14,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001) 


```


```{r astraltree, echo=F, fig.height=10, fig.width=9, warning=F, fig.cap="MQS tree made with ASTRAL III. Branch lengths represent coalescent units. Tip branches are an arbitrary length not calculated by ASTRAL III. Red numbers show local posterior probabilities"}

if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr)

tree <- read.tree(here("Data", "astral_ninety_coverage_supercontig7Apr.tre"))

tree$edge.length[is.na(tree$edge.length)] <- 0.35 #tip branches

tree <- root(tree, outgroup = "Oryza_sativa", resolve.root = T)

tree$edge.length[1] <- 0.1 #root branch

#Rename tips

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree$tip.label[tree$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Species[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

tree$tip.label[tree$tip.label == "Tetrarrhena laevis () "] <- "Tetrarrhena laevis"
tree$tip.label[tree$tip.label == "Microlaena stipoides () "] <- "Microlaena stipoides"



gg_tree <- ggtree(tree, right = T)
d <- gg_tree$data
d <- d[!d$isTip,]
d <- d[!d$label=="Root",]
da <- d[d$node %in% c(127, 184, 131, 185, 199, 132, 217, 136, 135, 137, 133),]
da$label[da$label == "1"] <- "1.00"


low <- gg_tree$data
lowl <- low[low$node %in% c(128,129,130,121,
                            122,123,124,125,126,
                            111,112,113,114,115,
                            116,117,118,119,120),]

lowl$label[lowl$label == "1"] <- "1.00"
lowl$label[lowl$label == "0.6"] <- "0.60"

ggtree(tree, right = T) +
    geom_treescale(x = -0.18, offset = 0.6, linesize = 0.6) +
    geom_tiplab(size = 2, colour = "black") +
    geom_rootedge(rootedge = 0.3) +
    geom_nodelab(data = da, aes(label = label),
                 size = 2.5,
                 nudge_x = -0.15,
                 nudge_y = -0.72,
                 colour = "#DF0101"
    ) +
    geom_nodelab(data = lowl, aes(label = label),
                 size = 2.5,
                 nudge_x = -0.15,
                 nudge_y = -0.72,
                 colour = "#DF0101"
    ) +
    
    #Major clades
    
    geom_cladelabel(node = 136, 
                    label = "Australian clade",
                    offset = 1.2,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.5,
                    offset.text = 0.04) +
    geom_cladelabel(node = 127, 
                    label = "Lowlands clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.2,
                    offset =1.4,
                    angle = 90,
                    hjust = 0.5,
                    offset.text = 0.04) +
    geom_cladelabel(node = 185, 
                    label = "Rehmannii clade",
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.2,
                    offset = 2.6,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.04) +
    geom_cladelabel(node = 199, 
                    label = "Ramosa clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.2,
                    offset = 2.5,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.04
    ) +
    geom_cladelabel(node = 137, 
                    label = "Rupestris clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.3,
                    offset = 2,
                    angle = 0,
                    #hjust = 0.5,
                    offset.text = 0.04) +
    geom_cladelabel(node = 133, 
                    label = "Dura clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 3,
                    extend = 0.3,
                    offset = 1,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.04)

```

## Species delimitation



### Rehmannii clade

#### Phylogenetic relationships



The ML and MQS analyses resolve three groups within the Rehmannii clade (Filiformis, Subspicata and Rehmannii groups) which correspond broadly to the existing subspecies of *E. rehmannii*, namely subsp. *filiformis*, subsp. *subspicata*, and subsp. *rehmannii*. Where the ML analysis identified all three groups as monophyletic with moderate to strong support (BS = 99, 77, 95, respectively), with the the Filiformis and Subspicata subgroups being sister lineages (BS = 97), the MQS analysis identifies only the Filiformis and Subspicata groups as monophyletic (LPP = 0.83, 0.96, respectively), the Rehmannii group being paraphyletic (LPP = 0.97, 0.85). Interestingly, a sample from near Agulhas, initially identified as *E. rehmannii* subsp. *filiformis* (1658), is placed in the Subspicata clade by both analyses.

#### GBS results

The three groups reconstructed by the phylogenetic analysis are largely corroborated by an sNMF analysis with three potential ancestral populations (K = 3), as inferences of ancestral population contribution generally coincides with the phylogenetic groups. Filiformis contains a predominantly homogeneous genetic composition across the sampled populations, with the exception of LW1. Occurring in the Cape Peninsula, LW1 shows potential admixture with both Subspicata and Rehmannii s.s populations. Collection 1658 was originally identified as an _E. rehmannii filiformis_, but the phylogenetic analysis placed it within the Subspicata group.  The sNMF results support this relationship, as 1658 shares a large proportion of its genomic information with the other Subspicata collections. Interestingly, the three populations sampled from the Agulhas area, 1658, 1659 and 1660, had reasonably different genetic profiles. In particular, 1660 shares a slightly greater proportion of ancestry with Filiformis than with Subspicata. 

The cross-entrophy criterion suggests that the optimal number of ancestral populations given the data is ten (K = 10). At this value of K there is strong population level genetic structuring, with almost every locality comprising of a distinct ancestral population. This is particularly apparent in the Filiformis group. Within Subspicata, there appears to be admixture between a Cape Peninsula population (LW3) and an Agulhas population (1659), but not within Agulhas populations (1658, 1659, 1660). Of the three clades, Rehmannii shows the most admixture between populations at this level. There is strong admixture between the Robinson's pass and Nature's valley populations (1649 and 1652), as well as some between Nature's Valley and Joubertina. However, it should be noted that the relationships presented for K = 10 were only reconstructed in 39% of the sNMF replications.


A PCOA of the GBS data largely agrees with sNMF findings. Each group formed a distinct cluster along PC1 and PC2 axes. Subspicata was somewhat dispersed, with each population forming a tight cluster within the greater cloud. The Filiformis and Subspicata groups showed also showed strong genetic variation along the PC3 and PC4 axes respectively.

#### Morphology

An LDA of the morphological data was able to separate the groups into discrete clusters. While Filiformis formed reasonably tight-knit group, Subspicata and Rehmannii s.s were more variable. A large proportion of the Subspicata individuals also grouped with Filiformis.  However, Rehmannii s.s tended to be taller, with a greater number of spikelets and an inflorescence that extended further above the leaves. Filiformis formed a reasonably cohesive group with a few outliers. This clade is more delicate than Rehmannii and Subspicata, with finer leaves, thinner culms and smaller and fewer spikelets.
\newpage

```{r rehmannii_phylogeny, echo=F, warning=F, fig.width=10, fig.height=9, fig.cap="Phylogenetic relationships of Rehmannii clade. A) ML tree constructed using IQtree. Red numbers show ultrafast bootstraps (BS). Branch lengths represent number of substitutions. B) MQS tree made with ASTRAL III. Branch lengths represent coalescent units (generations/ effective population size). Tip branches are an arbitrary length not calculated by ASTRAL III. Red numbers represent local posterior probabilities (LPP)"}


if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr)


#ML###

tree_ml <- read.tree(here("Data","boot_relaxed_clustering2_ninety_coverage_supercontig_21Apr.treefile"))
tree_ml <- root(tree_ml, outgroup = "Oryza_sativa", resolve.root = T)


#Rename tips
species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ml$tip.label[tree_ml$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

reh_clade <- extract.clade(tree_ml, 189)
reh_tree <- ggtree(reh_clade, right = T)
reh_dat <- reh_tree$data

reh_dat2 <- reh_dat[!reh_dat$isTip,]
reh_dat2 <- separate(reh_dat2, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)

ml_reh <- ggtree(reh_clade, size = 1, right = T) +
    geom_tiplab(size = 3.7, offset = 0.001) +
    geom_treescale(x = -0.002, width = 0.005, offset = 0.15, linesize = 0.6) +
    geom_rootedge(rootedge = 0.004) +
    geom_nodelab(data = reh_dat2, aes(label = bootstrap),
                 nudge_x = -0.0013,
                 nudge_y = -0.10,
                 colour = "#DF0101",
                 size = 3) +
    geom_cladelabel(node = 23, 
                    label = "Subspicata clade",
                    colour = "#F3A10E",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0015,
                    angle = 90,
                    align = T,
                    hjust = 0.5,
                    fontsize = 4.5) +
    geom_hilight(node=23, fill="#F3A10E", alpha = 0.15, extend = 0.03) +
    
    geom_cladelabel(node = 27, 
                    label = "Rehmannii s.s. clade",
                    colour = "#297AB1",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0015,
                    angle = 90,
                    hjust = 0.5,
                    align = T,
                    fontsize = 4.5) +
    geom_hilight(node=27, fill="#297AB1", alpha = 0.15, extendto = 0.046) +
    
    geom_cladelabel(node = 18, 
                    label = "Filiformis clade",
                    colour = "#0AB276",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0015,
                    angle = 90,
                    hjust = 0.5,
                    align = T,
                    fontsize = 4.5) +
    geom_hilight(node=18, fill="#0AB276", alpha = 0.15, extendto = 0.048)


#Astral ###


tree_ast <- read.tree(here("Data", "astral_ninety_coverage_supercontig7Apr.tre"))

tree_ast$edge.length[is.na(tree_ast$edge.length)] <- 0.35 #tip branches

tree_ast <- root(tree_ast, outgroup = "Oryza_sativa", resolve.root = T)

#Rename tips

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ast$tip.label[tree_ast$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

reh_ast_clade <- extract.clade(tree_ast, 185)

reh_tree_ast <- ggtree(reh_ast_clade, right = T)

reh_tree_ast <- flip(reh_tree_ast, 24,19)

reh_ast_dat <- reh_tree_ast$data
reh_ast_dat <- reh_ast_dat[!reh_ast_dat$isTip,]
reh_ast_dat$label[reh_ast_dat$label == "1"] <- "1.00"


ast_reh <- ggtree(reh_ast_clade, right = T, size = 1, ladderize = T) +
    geom_tiplab(size = 3.7, offset = 0.02) +
    geom_rootedge(rootedge = 0.05) +
    geom_treescale(x = -0.05, offset = 0.15, linesize = 0.6)+
    geom_nodelab(data = reh_ast_dat, aes(label = label),
                 nudge_x = -0.06,
                 nudge_y = -0.12,
                 colour = "#DF0101",
                 size = 3) +
    geom_cladelabel(node = 26, 
                    label = "Subspicata clade",
                    colour = "#F3A10E",
                    extend = 0.2,
                    offset = 1.4,
                    offset.text = 0.05,
                    angle = 90,
                    hjust = 0.5,
                    fontsize = 4.5,
                    align = T) +
    geom_hilight(node=26, fill="#F3A10E", alpha = 0.15, extendto = 2.15) +
    
    geom_cladelabel(node = 17, 
                    label = "Rehmannii s.s. clade",
                    colour = "#297AB1",
                    extend = 0.55,
                    offset = 1.4,
                    offset.text = 0.05,
                    angle = 90,
                    hjust = 0.7,
                    fontsize = 4.5,
                    align = T) +
    geom_cladelabel(node = 25, 
                    label = "",
                    colour = "#297AB1",
                    extend = 0.55,
                    offset = 1.4,
                    offset.text = 0.05,
                    angle = 90,
                    hjust = 0,
                    fontsize = 4.5,
                    align = T)+
    geom_hilight(node=25, fill="#297AB1", alpha = 0.15, extendto = 2) +
    geom_hilight(node=17, fill="#297AB1", alpha = 0.15, extendto = 2) +
    
    geom_cladelabel(node = 19, 
                    label = "Filiformis clade",
                    colour = "#0AB276",
                    extend = 0.3,
                    offset = 1.4,
                    offset.text = 0.05,
                    angle = 90,
                    hjust = 0.5,
                    fontsize = 4.5,
                    align = T) +
    geom_hilight(node=19, fill="#0AB276", alpha = 0.15, extendto = 1.9) 

ast_reh <- flip(ast_reh, 24,19)


ggpubr::ggarrange(ml_reh, ast_reh, labels = c("A", "B"), font.label = list(size = 19, face = "plain"), label.x = 0.07, label.y = 0.98)
```

```{r reh_ce, fig.cap="The cross-entropy criterion plotted for each value of K between K = 1 and K = 20. Red point shows K corresponding to number of phylogenetic clades, green point shows the K with the lowest cross-entropy criterion"}

means <- read.csv(here("Data", "rehmannii_ce.csv"))

plot(means$k, means$ce,
     ylab = "Cross-entropy", xlab = "Ancestral gene pool (K)", 
     pch = 19, cex = 1.2)

points(x = 3, y = means$ce[3],
       col = "red2",
       pch = 19, cex = 1.2)
points(x = 10, y = means$ce[10],
       col = "forest green",
       pch = 19, cex = 1.2)

```


```{r rehmannii_gbs, echo=F, warning=FALSE, cache=T, message=F, include = T, fig.height=6.4}

library(pophelper)
library(gridExtra)
library(label.switching)
library(tidyr)


clump_file3 <-readQ(here("Data", "calc.clumpp.merged.K3.txt"))


#population assignments
pop_names <- read.table(here("Data", "popmap_rehmannii_reordered"), stringsAsFactors=F)

clade_names <- c(
    rep("Filiformis", 16),
    rep("Subspicata", 20),
    rep("Rehmannii s.s", 12)
)


group_labs <- data.frame(collection = pop_names, clade = clade_names)
names(group_labs)[1] <- "collection"
group_labs <- separate(group_labs, 
                       col = collection, 
                       into = c("letter", "pop_id"),
                       sep = "A")
group_labs <- group_labs[,2:3]
group_labs$clade <- as.character(group_labs$clade)

my_colours <- c("#0AB276", #fili
                "#F3A10E", #sub
                "#297AB1" #reh
                ) 


p3 <- plotQ(clump_file3, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 4,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 3,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 3, 100/100, Mean(LnProb) = 0.000, Mean(similarity score) = 0.981",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)



#K = 10



clump_file <- readQ(here("Data", "calc.clumpp.merged.K10.txt"))

my_colours <- c(
  "#1D72F5","#DF0101","#77CE61", "#FF9326","#A945FF","#0089B2","#FDF060","#FFA6B2","#BFF217","#60D5FD"
)


p4 <- plotQ(clump_file, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 4,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 3,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 10, 39/100, Mean(LnProb) = 0.000, Mean(similarity score) = 0.972",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

ggpubr::ggarrange(arrangeGrob(p3$plot[[1]]), arrangeGrob(p4$plot[[1]]), nrow = 2)

```



```{r, include=F, fig.height=4}

library("vcfR")
library("adegenet")
library("dartR")
library("plotly")
library("here")

vcf <- read.vcfR(here("Data", "Final_lenient_rehmannii_28Apr.vcf"))

pop.assign<- read.csv(here("Data", "rehmannii_pop_assignment.csv"), header=T)
dat.pop.assign<-data.frame(pop.assign$Species_current, pop.assign$Pop)
colnames(dat.pop.assign)[colnames(dat.pop.assign)=="V1"] <- "pop.assign"

dat.pop.assign <- dat.pop.assign %>% 
  mutate(clade_labs =
         case_when(pop.assign.Species_current %in% c("E_rehmannii_filiformis") ~ "Filiformis", 
                   pop.assign.Species_current %in% c("E_rehmannii_rehmannii") ~ "Rehmannii s.s",
                   pop.assign.Species_current %in% c("E_rehmannii_subspicata") ~ "Subspicata"
                   )
)

dat.pop.assign$clade_labs[dat.pop.assign$pop.assign.Pop == "A1658"] <- "Subspicata"

#convert vcf to genlight
gl <- vcfR2genlight(vcf)

##Assign pop names so that R knows how to colour the PCoA points
strata(gl)<-data.frame(dat.pop.assign)
setPop(gl)<-~clade_labs

#Before PCoA, find & exclude the SNPs with missing data:
toRemove <- is.na(glMean(gl, alleleAsUnit = FALSE)) 
gl.na <- gl[, !toRemove]


my_colours <- c("#0AB276", #fili
                "#F3A10E", #sub
                "#297AB1" #reh
) 

my_theme <- theme_bw() +
  theme(panel.grid.minor = element_blank(),
        panel.grid = element_blank(),
        legend.key = element_rect(fill = NA, color = "grey"),
        legend.background = element_blank()
  )


pc<-gl.pcoa(gl.na, nfactors=5) 
pcoa_plot1 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=1, yaxis=2)

p1 <- pcoa_plot1 + my_theme +
          scale_colour_manual(values = my_colours, name = " ") +
          geom_point(size = 4)

pcoa_plot2 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=3, yaxis=4)

p2 <- pcoa_plot2 + my_theme +
  scale_colour_manual(values = my_colours, name = " ") +
  geom_point(size = 4) 


```

```{r pcoa_reh, echo=F, warning=FALSE, cache=T, message=F, fig.height=7}

ggpubr::ggarrange(p1, p2, nrow = 2, align = "hv", labels = c("A", "B"), font.label = list(size = 18, face = "plain"), label.x = 0.07, label.y = 0.98)


```


```{r reh_lda}
library(MASS)
library(dplyr)


trait_data <- read.csv(here("Data", "trait_dataV3.csv"), sep = ",")
trait_data$Collection <- as.character(trait_data$Collection)

trait_data <- trait_data %>% 
    filter(Species == "E_rehmannii_filiformis" | 
               Species == "E_rehmannii_rehmannii" |
               Species == "E_rehmannii_subspicata" ) %>% 
    droplevels() %>% 
    mutate(Clade = case_when(Collection %in% c("1649", "1652", "1653", "1647") ~ "Rehmannii s.s",
                             Collection %in% c( "1571", "1591", "1592", 
                                                "1615", "1567", "1", 
                                                "4", "1583") ~ "Filiformis",
                             Collection %in% c("1658","1660", "1590", "1659", "3") ~ "Subspicata"
    )
    )

trait_data <- trait_data %>% 
    mutate(inflor_position = Plant_height/Veg_height,
           relative_sheath_length = Sheath_length/Internode_dist,
           relative_leaf_length = Leaf_length/Sheath_length,
           glume_enclosure = Glume_length_outer/Spikelet_length,
           glume_ratio = Glume_length_outer/Glume_length_inner,
           spikelet_shape = Spikelet_length/Spikelet_width,
           lemma_ratio = Lemma_outer_length/Lemma_inner_length)

trait_data <- trait_data[, c(1:8, 11:22, 25:32)]
trait_data <- trait_data[, -c(4,8,14,15,16,18,19)]

my_colours <- c("#0AB276", "#297AB1", "#F3A10E")

my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          axis.title = element_text(size = 12)
         
    )


scale_reh <- data.frame(Clade = as.character(trait_data$Clade), scale(trait_data[, c(4:13, 15:21)]))

lda_rehmannii <- lda(scale_reh$Clade ~ ., data = scale_reh)

lda.data <- cbind(scale_reh, predict(lda_rehmannii)$x)

ggplot(lda.data, aes(LD1, LD2)) +
    geom_point(aes(color = Clade), size = 3) +
    scale_colour_manual(values = my_colours) +
    my_theme

```

```{r reh_boxplots, fig.height=8, message=F}
my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank()
    )

univariate_boxplot_trait_data <- function(x, y) {
    ggplot(trait_data, aes_string(x, y)) +
        geom_boxplot(fill = my_colours,
                     width = 0.6) +
        my_theme
}


#a <- univariate_boxplot_trait_data("Clade", "Plant_height")
#b <- univariate_boxplot_trait_data("Clade", "Veg_height")
#c <- univariate_boxplot_trait_data("Clade", "Leaf_width")
#d <- univariate_boxplot_trait_data("Clade", "Leaf_length")
#e <- univariate_boxplot_trait_data("Clade", "Sheath_length")
#f <- univariate_boxplot_trait_data("Clade", "Internode_dist")
#g <- univariate_boxplot_trait_data("Clade", "Culm_diam")
#h <- univariate_boxplot_trait_data("Clade", "Inflor_length")
#j <- univariate_boxplot_trait_data("Clade", "Spikelet_no")
#k <- univariate_boxplot_trait_data("Clade", "Pedicel_length_epu")
#l <- univariate_boxplot_trait_data("Clade", "Glume_length_outer")
#m <- univariate_boxplot_trait_data("Clade", "Glume_length_inner")
#ma <- univariate_boxplot_trait_data("Clade", "Lemma_inner_length")+ theme(axis.text.x = #element_text(angle = 40, vjust = 0.9, hjust = 1))
#mb <- univariate_boxplot_trait_data("Clade", "Lemma_outer_length")+ theme(axis.text.x = #element_text(angle = 40, vjust = 0.9, hjust = 1))
#n <- univariate_boxplot_trait_data("Clade", "Spikelet_length") + theme(axis.text.x = #element_text(angle = 40, vjust = 0.9, hjust = 1))
#z <- univariate_boxplot_trait_data("Clade", "Spikelet_width") +
#  theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
#
#ggpubr::ggarrange(a,b,c,d,e,f,g,h,j,k,l,m,ma,mb,n,z,q, 
#                  ncol = 4, nrow = 4, 
#                  align = 'v',
#                  heights = c(1,1,1,1.4))



a <- univariate_boxplot_trait_data("Clade", "inflor_position")
b <- univariate_boxplot_trait_data("Clade", "Veg_height")
c <- univariate_boxplot_trait_data("Clade", "Leaf_width")
d <- univariate_boxplot_trait_data("Clade", "Leaf_length")
#e <- univariate_boxplot_trait_data("Clade", "Sheath_length")
f <- univariate_boxplot_trait_data("Clade", "Internode_dist")
e <- univariate_boxplot_trait_data("Clade", "relative_sheath_length")
ea <- univariate_boxplot_trait_data("Clade", "relative_leaf_length")
g <- univariate_boxplot_trait_data("Clade", "Culm_diam")
h <- univariate_boxplot_trait_data("Clade", "Inflor_length")
j <- univariate_boxplot_trait_data("Clade", "Spikelet_no")
k <- univariate_boxplot_trait_data("Clade", "Pedicel_length_epu")
n <- univariate_boxplot_trait_data("Clade", "Spikelet_length") + theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
#l <- univariate_boxplot_trait_data("Clade", "Glume_length_outer")
l <- univariate_boxplot_trait_data("Clade", "glume_enclosure")
m <- univariate_boxplot_trait_data("Clade", "glume_ratio")
#m <- univariate_boxplot_trait_data("Clade", "Glume_length_inner")
#ma <- univariate_boxplot_trait_data("Clade", "Lemma_inner_length")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
ma <- univariate_boxplot_trait_data("Clade", "spikelet_shape")
mc <- univariate_boxplot_trait_data("Clade", "Spikelet_no")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
#mb <- univariate_boxplot_trait_data("Clade", "Lemma_outer_length")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
mb <- univariate_boxplot_trait_data("Clade", "lemma_ratio")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
z <- univariate_boxplot_trait_data("Clade", "Striation_no")

#z <- univariate_boxplot_trait_data("Clade", "Spikelet_width") +
  #theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))

ggpubr::ggarrange(c,d, NULL, b,a,e,ea, f,g,h,j,k,l,m,ma,mc, mb,n,q,z, 
                  ncol = 3, nrow = 6, 
                  align = 'v',
                  heights = c(1,1,1,1,1,1.4))
```


### Ramosa

The ML and MQS phylogenies resolve the Ramosa clade into three groups, Aphylla, Eastern ramosa and Western ramosa, in contrast to the existing taxonomy of two subspecies within _E. ramosa_. In both trees, Eastern ramosa is resolved as sister to the Aphylla and Western ramosa groups. Eastern ramosa group forms a well-supported clade within the MQS tree (LPP = 0.88), but consists of a grade in the ML tree. The Eastern ramosa group group consists of specimens collected along the Langeberg and Swartberg mountain ranges. The Aphylla group is monophyletic with strong support in both trees (BS = 100, LPP = 0.95), and contains collections from the Cape Peninsula to Betty's Bay. Within the Aphylla group the topology varies between the trees, with some poorly resolved internal nodes in the MQS tree. The Western ramosa group contains the remaining taxa, which mainly occur inland in the mid and western regions of the Western Cape. This group forms a grade in the MQS tree, with several poorly supported internal nodes.  _E. ramosa aphylla_ and _E. ramosa ramosa_ specimens that were sampled from the same localities in Milner peak and Limietberg are not recovered as sisters in either tree, suggesting that there may be multiple taxa with the Western ramosa group.

While the phylogenetic trees resolved three potential groups within Ramosa, the admixture results showed limited support for three ancestral populations (K = 3) that correspond to the phylogenetic groups. While all Aphylla individuals were assigned to the same ancestral population with high probability, most Western Ramosa individuals shared a large proportion of genomic information with the Aphylla group. A population from Milner peak (1624) was the exception, as all individuals from this locality were assigned to a largely distinct genetic cluster. However, this cluster also was present at low proportions with the other Western ramosa populations.  The Eastern ramosa populations were characterised by a shared ancestral gene pool, although several populations displayed admixture with the dominant gene pool in the Aphylla group.

The cross-entrophy criterion suggests that eight ancestral populations (K = 8) is the best fit for the dataset. At this level, the Aphylla group shows strong admixture between populations and consists of an ancestral gene pool that is not dominant in the other ramosa groups, supporting the phylogenetic evidence the Aphylla group is a cohesive clade. Conversely, the Western ramosa group shows strong substructure, as most populations are derived from distinct ancestral gene pools. The exception is populations 1622 and 1609 which show admixture, and were also resolved as sister taxa in the phylogenetic analyses. The two populations from Milner peak (1622 and 1624) do not share genetic information, despite being collected within a few kilometres of each other. The genetic variability of the Western ramosa group is also displayed in the PCOA, as although the group forms a rough cluster along the PC1 and PC2 axes, individual populations group together tightly, and are highly disparate along PC3 and PC4 axes. The Eastern ramosa group also shows some substructure, as populations collected from the Langeberg are distinct from both each other, and from the Swartberg populations. However, this substructure appears to be less strong than that of the Western ramosa group, as Eastern ramosa shows a cohesive cluster along all PC axes.







```{r, echo=F, warning=F, fig.width=10, fig.height=9, fig.cap="Phylogenetic relationships of Ramosa clade. A) ML tree constructed using IQtree. Red numbers show ultrafast bootstraps (BS). Branch lengths represent number of substitutions. B) MQS tree made with ASTRAL III. Branch lengths represent coalescent units (generations/ effective population size). Tip branches are an arbitrary length not calculated by ASTRAL III. Red numbers represent local posterior probabilities (LPP)"}

#Ramosa

if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr)


#ML####

#Rename tips
species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ml$tip.label[tree_ml$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

ram_clade <- extract.clade(tree_ml, 171)
ram_tree <- ggtree(ram_clade, right = T)
ram_dat <- ram_tree$data

ram_dat2 <- ram_dat[!ram_dat$isTip,]
ram_dat2 <- separate(ram_dat2, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)


ml_ram <- ggtree(ram_clade, size = 1, right = T) +
    geom_tiplab(size = 3.7, offset = 0.001) +
    geom_treescale(x = -0.002, width = 0.005, offset = 0.15, linesize = 0.6) +
    geom_rootedge(rootedge = 0.004) +
    geom_nodelab(data = ram_dat2, aes(label = bootstrap),
                 nudge_x = -0.0017,
                 nudge_y = -0.17,
                 colour = "#DF0101",
                 size = 3) +
    geom_cladelabel(node = 31, 
                    label = "Aphylla clade",
                    colour = "#00A0B0",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0015,
                    angle = 90,
                    align = T,
                    hjust = 0.5,
                    fontsize = 4.5) +
    geom_hilight(node=31, fill="#00A0B0", alpha = 0.2, extend = 0.03) +
    geom_cladelabel(node = 23, 
                    label = "Western ramosa clade",
                    colour = "#9BC53D",
                    extend = 0.35,
                    offset = 0.035,
                    offset.text = 0.0015,
                    angle = 90,
                    align = T,
                    hjust = 0.5,
                    fontsize = 4.5) +
    geom_hilight(node=23, fill="#9BC53D", alpha = 0.2, extend = 0.03) +
   
    geom_cladelabel(node = 37, 
                    label = "Eastern ramosa clade",
                    colour = "#CC333F",
                    extend = 1.5,
                    offset = 0.035,
                    offset.text = 0.0015,
                    angle = 90,
                    align = T,
                    hjust = 0.62,
                    fontsize = 4.5) +
    geom_cladelabel(node = 35, 
                    label = "",
                    colour = "#CC333F",
                    extend = 0.45,
                    offset = 0.035,
                    offset.text = 0.0015,
                    angle = 90,
                    align = T,
                    hjust = 0.5,
                    fontsize = 4.5) +
    geom_hilight(node=36, fill="#CC333F", alpha = 0.2, extendto = 0.05) +
    geom_hilight(node=35, fill="#CC333F", alpha = 0.2, extendto = 0.05)
    

#Astral ####


#Rename tips

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ast$tip.label[tree_ast$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

ram_ast_clade <- extract.clade(tree_ast, 199)

ram_tree_ast <- ggtree(ram_ast_clade, right = T)
ram_tree_ast <- flip(ram_tree_ast,22,28)
ram_tree_ast <- flip(ram_tree_ast,23,24)

ram_ast_dat <- ram_tree_ast$data
ram_ast_dat <- ram_ast_dat[!ram_ast_dat$isTip,]
ram_ast_dat[ram_ast_dat == "1"] <- "1.00"
ram_ast_dat[ram_ast_dat == "0.7"] <- "0.70"
ram_ast_dat[ram_ast_dat == "0.4"] <- "0.40"

ast_ram <- ggtree(ram_ast_clade, right = T, size = 1, ladderize = T) +
    geom_tiplab(size = 3.7, offset = 0.02) +
    geom_rootedge(rootedge = 0.05) +
    geom_treescale(x = -0.05, width = 0.08, offset = 0.15, linesize = 0.6)+
    geom_nodelab(data = ram_ast_dat, aes(label = label),
                 nudge_x = -0.07,
                 nudge_y = -0.17,
                 colour = "#DF0101",
                 size = 3) +
    geom_cladelabel(node = 24, 
                    label = "Aphylla clade",
                    colour = "#00A0B0",
                    extend = 0.2,
                    offset = 1.4,
                    offset.text = 0.05,
                    angle = 90,
                    hjust = 0.5,
                    fontsize = 4.5,
                    align = T) +
    geom_hilight(node=24, fill="#00A0B0", alpha = 0.2, extendto = 1.8) +
    geom_cladelabel(node = 34, 
                    label = "Eastern ramosa clade",
                    colour = "#CC333F",
                    extend = 0.5,
                    offset = 1.4,
                    offset.text = 0.05,
                    angle = 90,
                    hjust = 0.5,
                    fontsize = 4.5,
                    align = T) +
    geom_hilight(node=34, fill="#CC333F", alpha = 0.2, extendto = 1.7)+
    geom_cladelabel(node = 28, 
                    label = "Western ramosa clade",
                    colour = "#9BC53D",
                    extend = 0.8,
                    offset = 1.4,
                    offset.text = 0.05,
                    angle = 90,
                    hjust = 0.4,
                    fontsize = 4.5,
                    align = T) +
    geom_cladelabel(node = 23, 
                    label = "",
                    colour = "#9BC53D",
                    extend = 0.4,
                    offset = 1.4,
                    offset.text = 0.05,
                    angle = 90,
                    hjust = 0.5,
                    fontsize = 4.5,
                    align = T) +
    geom_hilight(node=28, fill="#9BC53D", alpha = 0.2, extendto = 1.5) +
    geom_hilight(node=23, fill="#9BC53D", alpha = 0.2, extendto = 1.5)
ast_ram <- flip(ast_ram, 22, 28)
ast_ram <- flip(ast_ram,23,24)

ggpubr::ggarrange(ml_ram, ast_ram, labels = c("A", "B"), font.label = list(size = 19, face = "plain"), label.x = 0.07, label.y = 0.98)


```



```{r ram_ce, fig.cap="The cross-entropy criterion plotted for each value of K between K = 1 and K = 20. Red point shows K corresponding to number of phylogenetic clades, green point shows the K with the lowest cross-entropy criterion"}

means <- read.csv(here("Data", "ramosa_ce.csv"))

plot(means$k, means$ce,
     ylab = "Cross-entropy", xlab = "Ancestral gene pool (K)", 
     pch = 19, cex = 1.2)

points(x = 3, y = means$ce[3],
       col = "red2",
       pch = 19, cex = 1.2)
points(x = 8, y = means$ce[8],
       col = "forest green",
       pch = 19, cex = 1.2)

```


```{r ram_gbs, echo=F, warning=FALSE, cache=F, message=F, include = T, fig.height=6.4}

library(pophelper)
library(gridExtra)
library(label.switching)
library(tidyr)


clump_file <-readQ(here("Data", "calc.clumpp.merged.K3_ram.txt"))

#population assignments
pop_names <- read.table(here("Data", "popmap_ramosa_reordered"), stringsAsFactors=F)

clade_names <- c(
    rep("Aphylla", 12),
    rep("Western ramosa", 20),
    rep("Eastern ramosa", 16)
)

group_labs <- data.frame(collection = pop_names, clade = clade_names)
names(group_labs)[1] <- "collection"
group_labs <- separate(group_labs, 
                       col = collection, 
                       into = c("letter", "pop_id"),
                       sep = "A")
group_labs <- group_labs[,2:3]
group_labs$clade <- as.character(group_labs$clade)

my_colours <- c("#00A0B0", #aphylla
                "#CC333F", #E ram
                "#9BC53D" #W ram
) 


p1 <- plotQ(clump_file, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 4,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 4,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 3, 93/100, Mean(LnProb) = 0.000, Mean(similarity score) = 0.982",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

#K = 8


clump_file2 <- readQ(here("Data", "calc.clumpp.merged.K8.txt"))

my_colours <- c(
    "#1D72F5","#DF0101","#77CE61", "#FF9326","#A945FF","#0089B2","#FDF060","#FFA6B2","#BFF217","#60D5FD"
)



p2 <- plotQ(clump_file2, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 4,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 4,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 8, 60/100, Mean(LnProb) = 0.000, Mean(similarity score) = 0.958",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

ggpubr::ggarrange(arrangeGrob(p1$plot[[1]]), arrangeGrob(p2$plot[[1]]), nrow = 2)
```


```{r ram_pcoa, include=F}
library("vcfR")
library("adegenet")
library("dartR")
library("plotly")

vcf <- read.vcfR(here("Data", "FinalRamosa_lenient27Apr.vcf"))

pop.assign<- read.csv(here("Data", "ramosa_population_assignment.csv"), header=T)
dat.pop.assign<-data.frame(pop.assign$Species_current, pop.assign$Pop)
colnames(dat.pop.assign)[colnames(dat.pop.assign)=="V1"] <- "pop.assign"

dat.pop.assign <- dat.pop.assign %>% 
    mutate(clade_labs = c(rep("Aphylla", 8), rep("Western ramosa", 8), 
                          rep("Aphylla", 4), rep("Western ramosa", 12),
                          rep("Eastern ramosa",16)) 
    )

#convert vcf to genlight
gl <- vcfR2genlight(vcf)

##Assign pop names so that R knows how to colour the PCoA points
strata(gl)<-data.frame(dat.pop.assign)
setPop(gl)<-~clade_labs


#Before PCoA, find & exclude the SNPs with missing data:
toRemove <- is.na(glMean(gl, alleleAsUnit = FALSE)) 
which(toRemove) 
gl.na <- gl[, !toRemove]

my_colours <- c("#00A0B0", #aphylla
                "#9BC53D", #W ram
                "#CC333F" #E ram
               
)

my_theme <- theme_bw() +
  theme(panel.grid.minor = element_blank(),
        panel.grid = element_blank(),
        legend.key = element_rect(fill = NA, color = "grey"),
        legend.background = element_blank()
  )


pc<-gl.pcoa(gl.na, nfactors=5) 
pcoa_plot1 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=1, yaxis=2)

p1 <- pcoa_plot1 + my_theme +
          scale_colour_manual(values = my_colours, name = " ") +
          geom_point(size = 4)

pcoa_plot2 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=3, yaxis=4)

p2 <- pcoa_plot2 + my_theme +
  scale_colour_manual(values = my_colours, name = " ") +
  geom_point(size = 4) 


```

```{r pcoa_ram, echo=F, warning=FALSE, cache=T, message=F, fig.height=7}

ggpubr::ggarrange(p1, p2, nrow = 2, align = "hv", labels = c("A", "B"), font.label = list(size = 18, face = "plain"), label.x = 0.07, label.y = 0.98)


```


```{r ram_lda}
my_colours <- c("#00A0B0", "#DF0101",  "#9BC53D")


trait_data <- read.csv(here("Data", "trait_dataV3.csv"), sep = ",")
trait_data$Collection <- as.character(trait_data$Collection)

trait_data <- trait_data %>% 
    filter(Species == "E_ramosa_aphylla" | 
               Species == "E_ramosa_ramosa") %>% 
    droplevels() %>% 
    
    mutate(Clade = 
              case_when(Collection %in% c("1566", "1570", "1601", "1576") ~ "Aphylla",
                        Collection %in% c("1656", "1654", "1635", "1643", "1645") ~ "East. ramosa",
                        Collection %in% c("1606", "1560", "1585", "1609", "1611",
                                          "1588", "1626", "1622", "1624") ~ "West. ramosa"
              )
    )


trait_data <- trait_data %>% 
    mutate(inflor_position = Plant_height/Veg_height,
           relative_sheath_length = Sheath_length/Internode_dist,
           relative_leaf_length = Leaf_length/Sheath_length,
           glume_enclosure = Glume_length_outer/Spikelet_length,
           glume_ratio = Glume_length_outer/Glume_length_inner,
           spikelet_shape = Spikelet_length/Spikelet_width,
           lemma_ratio = Lemma_outer_length/Lemma_inner_length)

trait_data <- trait_data[, c(1:8, 11:22, 25:32)]
trait_data <- trait_data[, -c(4,8,14,15,16,18,19)]


scale_ram <- data.frame(Clade = as.character(trait_data$Clade), scale(trait_data[,c(4:13, 15:21)]))

lda_ramosa <- lda(scale_ram$Clade ~ ., data = scale_ram)

lda.data <- cbind(scale_ram, predict(lda_ramosa)$x)

ggplot(lda.data, aes(LD1, LD2)) +
    geom_point(aes(color = Clade), size = 2) +
    scale_colour_manual(values = my_colours) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))
```


```{r ram_boxplots, fig.height=8, message=F}
my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank()
    )

#a <- univariate_boxplot_trait_data("Clade", "Plant_height")
a <- univariate_boxplot_trait_data("Clade", "inflor_position")
b <- univariate_boxplot_trait_data("Clade", "Veg_height")
c <- univariate_boxplot_trait_data("Clade", "Leaf_width")
d <- univariate_boxplot_trait_data("Clade", "Leaf_length")
#e <- univariate_boxplot_trait_data("Clade", "Sheath_length")
f <- univariate_boxplot_trait_data("Clade", "Internode_dist")
e <- univariate_boxplot_trait_data("Clade", "relative_sheath_length")
ea <- univariate_boxplot_trait_data("Clade", "relative_leaf_length")
g <- univariate_boxplot_trait_data("Clade", "Culm_diam")
h <- univariate_boxplot_trait_data("Clade", "Inflor_length")
j <- univariate_boxplot_trait_data("Clade", "Spikelet_no")
k <- univariate_boxplot_trait_data("Clade", "Pedicel_length_epu")
n <- univariate_boxplot_trait_data("Clade", "Spikelet_length") + theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
#l <- univariate_boxplot_trait_data("Clade", "Glume_length_outer")
l <- univariate_boxplot_trait_data("Clade", "glume_enclosure")
m <- univariate_boxplot_trait_data("Clade", "glume_ratio")
#m <- univariate_boxplot_trait_data("Clade", "Glume_length_inner")
#ma <- univariate_boxplot_trait_data("Clade", "Lemma_inner_length")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
ma <- univariate_boxplot_trait_data("Clade", "spikelet_shape")
mc <- univariate_boxplot_trait_data("Clade", "Spikelet_no")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
#mb <- univariate_boxplot_trait_data("Clade", "Lemma_outer_length")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
mb <- univariate_boxplot_trait_data("Clade", "lemma_ratio")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
z <- univariate_boxplot_trait_data("Clade", "Striation_no")

#z <- univariate_boxplot_trait_data("Clade", "Spikelet_width") +
  #theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))

ggpubr::ggarrange(c,d, NULL, b,a,e,ea, f,g,h,j,k,l,m,ma,mc, mb,n,q,z, 
                  ncol = 3, nrow = 6, 
                  align = 'v',
                  heights = c(1,1,1,1,1,1.4))

```



### Rupestris

The Rupestris clade consists of two species, _E. setacea_ and _E. rupestris_, that are highly polyphyletic. The clade is characterised by extremely short internal branch lengths and poor internal support. Although the weak internal resolution results in topological incongruencies between the trees, there are four major subclades that are consistently recovered with good support in both analyses, which in turn can be divided into smaller groups. The first two major subclades are consist of the Eastern rupestris group (BS = 100, LPP = 1.0) and the Rupestris s.s group (BS = 100, LPP = 1.0) respectively, demonstrating that the existing subspecies _E. rupestris_ subsp. _rupestris_ is not monophyletic. Another major subclade consists of the Fernkloof setacea, Uniflora, and the Restiod tricostata groups (BS = 99, LPP = 0.74), with Uniflora and Restiod tricostata recovered as sisters in both trees (BS = 100, LPP = 0.72). The final major subclade encompasses the Setacea, Dodii, Leafy tricostata and Scabra groups (BS = 97, LPP = 0.68), although the position of these groups are incongruent between the trees. The ML tree also places Rupestris s.s within this subclade with weak support (BS = 52).

There are nine smaller groups within the subclades. The Fernkloof group consists of _E. setacea setacea_ and _E. setacea disticha_ taxa collected from Fernkloof reserve in Hermanus, and well as _E. setacea disticha_ from the nearby Buffelstalberg close to Betty's Bay. This group shows incongruencies between the two trees, forming a grade in the ML tree and a poorly supported clade in the MQS tree (LPP = 0.37). The main uncertainty arises from the position of the Buffelstalberg  _E. setacea disticha_, which is sister to the rest of the group in the ML tree (BS = 99), but is reconstructed within the group by the MQS analyses, albeit with poor support (LPP = 0.45). In both trees, the populations 1600 and 1595 are sisters with strong support (BS = 100, LPP = 1.00), and the _E. setacea disticha_ from Fernkloof forms a clade with two other Fernkloof collections identified as _E. setacea setacea_ (BS = 98, LPP = 0.88). The other two groups in this subclade, Uniflora and Restiod tricostata, have similar topologies in both trees, and form well supported clades (BS = 100, LPP = 1.00 and BS = 100, LPP = 0.97 respectively). The Setacea group formed a large, poorly supported clade in in the MQS tree (LPP = 0.56), and a grade in the ML tree. The internal support within this group was poor in both trees, resulting in high incongruence between the two phylogenies. The ML tree also placed an individual identified as an _E. rupestris_ subsp. _tricostata_ (1657) within this clade. The Scabra group was resolved as strongly supported clade in both trees (BS = 100, LPP = 1.00), although an individual previously identified as _E. rupestris_ subsp. _tricostata_ (1640) was reconstructed as sister to the rest of the group. Similarly, the Dodii group was found to be monophyletic with a nearly identical internal topology in both analyses (BS = 96, LPP = 0.95). There was slightly weaker support for monophyly of the Leafy tricostata group in the MQS tree (LPP = 0.70), and the group formed a grade in ML tree.

The sNMF analysis largely supports the nine groups found by phylogenetic evidence. Dodii, Leafy tricostata, Restiod tricostata, Eastern rupestris, Uniflora and Rupestris s.s. populations show strong within-group admixture, but all have a distinct genomic composition. This is also displayed in the PCOA, where each group aggregates into cohesive clusters. The Setacea group shows some admixture the other groups, which is unsurprising given that it is one of the most widespread species in the Rupestris clade. The sNMF analysis also reveals a potential tenth group, a population from Wemmershoek near Paarl (1657), which was not well resolved in the phylogenetic analysis. Unexpectedly, Scabra has genomic information in common with multiple ancestral pools, with the greatest proportion shared with the Leafy tricostata group. However, there is little evidence that admixture has occurred between Scabra and the sympatric Setacea population (1633), and the Scabra group forms a distinct cluster in the PCOA. The Fernkloof group also has a complex genetic profile. Populations 1589 and 1597, both currently classified as *E. setacea disticha*, have a similar genomic composition, although no ancestral group dominates. Conversely, 1600, collected within a few kilometres of 1597, shows very few shared alleles with the other Fernkloof populations. The three Fernkloof collections do cluster together in the PCOA, albeit with some population level structure. 

The best number of ancestral populations within the Rupestris clade according to the cross-entrophy criterion was k = 19. At this level, almost every group shows substructure, although there is some evidence of within-group admixture in the Setacea, Restiod tricostata and Dodii groups. Notably, within the Fernkloof group, the two *E. setacea disticha* (1589, 1597) populations share genomic information with each other but not with 1600.



#### Morphology

Based on the sNMF and phylogenetic analysis, I split the Fernkloof group into two, with Fernkloof A consisting of the disticha forms, and Fernkloof B consisting of collections 1595 and 1600.  A linear discriminant analysis (LDA) of the morphological data shows that both the Uniflora and Restiod tricostata groups have distinctive morphologies compared to the remainder of the Rupestris clade. Restiod tricostata is characterised by large internode distances, long sheathes and a small glume ratio, while Uniflora is a tall, gracile plant with thin culms and fine leaves. In order to investigate the fine scale variation between the remaining groups, Restiod tricostata and Uniflora were removed from the analysis. Three large clusters form with LD1 and LD2, consisting of Dodii and Leafy tricostata, Scabra and the rupestris groups, and Setacea and the Fernkloof groups. Dodii and Leafy tricostata mainly differ in the number of spikelets, with most other traits having similar values, although Leafy tricostata tends towards a more robust form in both vegetative and floral traits than Dodii. While Setacea and the Fernkloof groups



```{r, echo=F, warning=F, fig.width=10, fig.height=9, fig.cap="Phylogenetic relationships of Ramosa clade. A) ML tree constructed using IQtree. Red numbers show ultrafast bootstraps (BS). Branch lengths represent number of substitutions. B) MQS tree made with ASTRAL III. Branch lengths represent coalescent units (generations/effective population size). Tip branches are an arbitrary length not calculated by ASTRAL III. Red numbers represent local posterior probabilities (LPP)"}

#Rupestris minorclades

if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr)


#ML###


#Rename tips
species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ml$tip.label[tree_ml$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

rup_clade <- extract.clade(tree_ml, 120)

rup_tree <- ggtree(rup_clade, right = T)
rup_dat <- rup_tree$data

rup_dat2 <- rup_dat[!rup_dat$isTip,]
rup_dat2 <- separate(rup_dat2, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)

ml_rup <- ggtree(rup_clade, size = 1, right = T) +
    geom_tiplab(size = 3.7, offset = 0.001) +
    geom_treescale(x = -0.002, width = 0.005, offset = 0.15, linesize = 0.6) +
    geom_rootedge(rootedge = 0.004) +
    geom_nodelab(data = rup_dat2, aes(label = bootstrap),
                 nudge_x = -0.0019,
                 nudge_y = -0.33,
                 colour = "#DF0101",
                 size = 3) +
    geom_cladelabel(node = 95, 
                    label = "East. rupestris",
                    colour = "#1D72F5",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=95, fill="#1D72F5", alpha = 0.15, extend = 0.031) +
    
    geom_cladelabel(node = 89, 
                    label = "Fernkloof",
                    colour = "#DF0101",
                    extend = 0.6,
                    offset = 0.035,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.54,
                    fontsize = 4) +
    geom_cladelabel(node = 94, 
                    label = "",
                    colour = "#DF0101",
                    extend = 0.45,
                    offset = 0.035,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=89, fill="#DF0101", alpha = 0.15, extendto = 0.045) +
    geom_hilight(node=94, fill="#DF0101", alpha = 0.15, extendto = 0.045) +
    
    geom_cladelabel(node = 84, 
                    label = "Uniflora",
                    colour = "#9471B4",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=84, fill="#9471B4", alpha = 0.15, extend = 0.0355) +
    geom_cladelabel(node = 85, 
                    label = "Restiod tricost.",
                    colour = "#FF9326",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=85, fill="#FF9326", alpha = 0.15, extend = 0.037) +
    
    geom_cladelabel(node = 76, 
                    label = "Dodii",
                    colour = "#77CE61",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=76, fill="#77CE61", alpha = 0.15, extend = 0.032) +
    
    geom_cladelabel(node = 55, 
                    label = "Setacea",
                    colour = "#0089B2",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=55, fill="#0089B2", alpha = 0.15, extendto = 0.05) +
    geom_hilight(node=53, fill="#0089B2", alpha = 0.15, extendto = 0.05) +
    
    geom_cladelabel(node = 73, 
                    label = "Scabra",
                    colour = "#F2B950",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=73, fill="#F2B950", alpha = 0.15, extend = 0.031) +
    
    geom_cladelabel(node = 71, 
                    label = "Leafy tricost.",
                    colour = "#027368",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=71, fill="#027368", alpha = 0.15, extend = 0.034) +
    
    geom_cladelabel(node = 67, 
                    label = "Rupestris s.s",
                    colour = "#CC1577",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.001,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=67, fill="#CC1577", alpha = 0.15, extend = 0.03) +
    geom_cladelabel(node = 89, 
                    label = "blank",
                    colour = "#DF0101",
                    extend = 0.71,
                    offset = 0.049,
                    offset.text = 0.0012,
                    angle = 90,
                    align = T,
                    alpha = 0,
                    #hjust = 0.54,
                    fontsize = 4)

#Astral ###########



#Rename tips

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ast$tip.label[tree_ast$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

tree_ast$tip.label[tree_ast$tip.label == "uniflora (1572) Cape Peninsula"] <- "uniflora (1572) Cape Pen."

tree_ast$tip.label[tree_ast$tip.label == "tricostata (1594 repeat) Fernkloof"] <- "tricostata (1594 rep.) Fernkloof"

rup_ast_clade <- extract.clade(tree_ast, 137)

rup_tree_ast <- ggtree(rup_ast_clade, right = T)
rup_ast_dat <- rup_tree_ast$data
rup_ast_dat <- rup_ast_dat[!rup_ast_dat$isTip,]
rup_ast_dat[rup_ast_dat == "1"] <- "1.00"
rup_ast_dat[rup_ast_dat == "0.4"] <- "0.40"
rup_ast_dat[rup_ast_dat == "0.6"] <- "0.60"
rup_ast_dat[rup_ast_dat == "0.7"] <- "0.70"


ast_rup <- ggtree(rup_ast_clade, right = T, size = 1, ladderize = T) +
    geom_tiplab(size = 3.7, offset = 0.02) +
    geom_rootedge(rootedge = 0.05) +
    geom_treescale(x = -0.05, width = 0.10, offset = 0.15, linesize = 0.6)+
    geom_nodelab(data = rup_ast_dat, aes(label = label),
                 nudge_x = -0.15,
                 nudge_y = -0.35,
                 colour = "#DF0101",
                 size = 3) +
    geom_cladelabel(node = 50, 
                    label = "Rupestris s.s",
                    colour = "#CC1577",
                    extend = 0.2,
                    offset = 1.5,
                    offset.text = 0.05,
                    #angle = 0,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=50, fill="#CC1577", alpha = 0.15, extend = 2.2) +
    
    geom_cladelabel(node = 57, 
                    label = "Uniflora",
                    colour = "#9471B4",
                    extend = 0.2,
                    offset = 1.95,
                    offset.text = 0.05,
                    angle = 0,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=57, fill="#9471B4", alpha = 0.15, extend = 1.95) +
    
    geom_cladelabel(node = 58, 
                    label = "Restiod tricost. ",
                    colour = "#FF9326",
                    extend = 0.2,
                    offset = 1.5,
                    offset.text = 0.05,
                    angle = 0,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=58, fill="#FF9326", alpha = 0.15, extend = 2.3) +
    
    geom_cladelabel(node = 62, 
                    label = "Fernkloof",
                    colour = "#DF0101",
                    extend = 0.2,
                    offset = 1.5,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=62, fill="#DF0101", alpha = 0.15, extend = 2.2) +
    
    geom_cladelabel(node = 70, 
                    label = "East. rupestris",
                    colour = "#1D72F5",
                    extend = 0.2,
                    offset = 1.5,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=70, fill="#1D72F5", alpha = 0.15, extend = 2.2) +
    
    geom_cladelabel(node = 73, 
                    label = "Scabra",
                    colour = "#F2B950",
                    extend = 0.2,
                    offset = 1.5,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=73, fill="#F2B950", alpha = 0.15, extend = 2.2) +
    
    geom_cladelabel(node = 77, 
                    label = "Leafy tricost.",
                    colour = "#027368",
                    extend = 0.2,
                    offset = 1.5,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=77, fill="#027368", alpha = 0.15, extend = 2.1) +
    
    geom_cladelabel(node = 81, 
                    label = "Dodii",
                    colour = "#77CE61",
                    extend = 0.2,
                    offset = 1.5,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=81, fill="#77CE61", alpha = 0.15, extend = 2.2) +
    
    geom_cladelabel(node = 86, 
                    label = "Setacea",
                    colour = "#0089B2",
                    extend = 0.2,
                    offset = 1.5,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=86, fill="#0089B2", alpha = 0.15, extend = 2.5) +
    
    geom_cladelabel(node = 54, 
                    label = "Blank",
                    colour = "red",
                    extend = 0.21,
                    offset = 2.33,
                    offset.text = 0.05,
                    angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    alpha = 0,
                    align = T)
    
ggpubr::ggarrange(ml_rup, ast_rup, labels = c("A", "B"), font.label = list(size = 19, face = "plain"), label.x = 0.07, label.y = 0.98)



```


```{r rup_ce, fig.cap="The cross-entropy criterion plotted for each value of K between K = 1 and K = 25. Red point shows K corresponding to number of phylogenetic clades, green point shows the K with the lowest cross-entropy criterion"}

means <- read.csv(here("Data", "rupestris_ce.csv"))

plot(means$k, means$ce,
     ylab = "Cross-entropy", xlab = "Ancestral gene pool (K)", 
     pch = 19, cex = 1.2)

points(x = 9, y = means$ce[9],
       col = "red2",
       pch = 19, cex = 1.2)
points(x = 19, y = means$ce[19],
       col = "forest green",
       pch = 19, cex = 1.2)

```

```{r rup_gbs, echo=F, warning=FALSE, cache=F, message=F, include = T, fig.height=6.4}
library(pophelper)
library(gridExtra)
library(label.switching)
library(tidyr)


clump_file <-readQ(here("Data", "calc.clumpp.merged.K9.txt"))

pop_names <- read.table(here("Data", "popmap_rupestris_reordered"), stringsAsFactors=F)

clade_names <- c(
    rep("Setacea", 20),
    rep("Fernkloof", 11),
    rep("Scab", 4),
    rep("Uni", 4),
    rep("Dodii", 12),
    rep("Leafy tri.", 12),
    rep("Restiod tri.", 12),
    rep("Wem", 4),
    rep("Rup.ss", 8),
    rep("East.rup.", 8)
)

group_labs <- data.frame(collection = pop_names, clade = clade_names)
names(group_labs)[1] <- "collection"
group_labs <- separate(group_labs, 
                       col = collection, 
                       into = c("letter", "pop_id"),
                       sep = "B")
group_labs <- group_labs[,2:3]
group_labs$clade <- as.character(group_labs$clade)

my_colours <- c(
    "#60D5FD", #set
    "#027368", #leafy
    "#77CE61", #dodii
    "dark grey",
    "#1D72F5", #E. rup
    "#851D41", #rup
    "#DF0101", #Fernkloof
     #"#F2B950", #scab
    "#9471B4", #uni
    "#FF9326" #restiod
) 


p5 <- plotQ(clump_file, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 2.7,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 4,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 9, 83/100, Mean(LnProb) = 0.000, Mean(similarity score) = 0.976",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

#k = 19

clump_file2 <- readQ(here("Data", "calc.clumpp.merged.K19.txt"))

my_colours <- c("#1D72F5","#DF0101","#77CE61", "#FF9326","#A945FF","#0089B2","#FDF060","#FFA6B2","#BFF217","#60D5FD","#CC1577","#F2B950","#7FB21D","#EC496F","#326397","#B26314","#CCEBC5","#A4A4A4","#610B5E")



p2 <- plotQ(clump_file2, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 2.7,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 3,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 1.9,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 19, 61/100, Mean(LnProb) = 0.000, Mean(similarity score) = 0.708",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

ggpubr::ggarrange(arrangeGrob(p5$plot[[1]]), arrangeGrob(p2$plot[[1]]), nrow = 2)
```

```{r, include = F}

library("vcfR")
library("adegenet")
library("dartR")
library("plotly")
library(tidyverse)

#read.vcf
vcf <- read.vcfR(here("Data", "Final_rupestris_lenient24Apr.vcf"))

#assign pop identifiers
pop.assign<- read.csv(here("Data", "rupestris_pop_assignment.csv"), header=T)
dat.pop.assign<-data.frame(pop.assign$Species_current, pop.assign$Clade)


dat.pop.assign <- dat.pop.assign %>% 
    mutate(clade_labs =
               case_when(pop.assign.Clade %in% c("setacea") ~ "Setacea", 
                         pop.assign.Clade %in% c("rupestris_original") ~ "Rupestris s.s",
                         pop.assign.Clade %in% c("dodii_original") ~ "Dodii",
                         pop.assign.Clade %in% c("tricostata_original") ~ "Restiod tricost.",
                         pop.assign.Clade %in% c("fernkloof_setacea") ~ "Fernkloof",
                         pop.assign.Clade %in% c("uniflora") ~ "Uniflora",
                         pop.assign.Clade %in% c("milner_group") ~ "Leafy tricost.",
                         pop.assign.Clade %in% c("scabra") ~ "Scabra",
                         pop.assign.Clade %in% c("rupestris_swartberg") ~ "Eastern rupestris",
                         pop.assign.Clade %in% c("wemmershoek") ~ "Wemmershoek"
               )
    )


#convert vcf to genlight
gl <- vcfR2genlight(vcf)

##Assign pop names so that R knows how to colour the PCoA points
strata(gl)<-data.frame(dat.pop.assign)
setPop(gl)<-~clade_labs

#Before PCoA, find & exclude the SNPs with missing data:
toRemove <- is.na(glMean(gl, alleleAsUnit = FALSE)) 
which(toRemove) 
gl.na <- gl[, !toRemove]
pop(gl.na)

my_colours <- c(
    "#60D5FD", #set
    "#851D41", #rup
    "#77CE61", #dodii
    "#FF9326", #restiod
    "#DF0101", #Fernkloof
    "#9471B4", #uni
    "#027368", #leafy
    "#F2B950", #scab
    "#1D72F5", #E. rup
    "grey"
)  

my_theme <- theme_bw() +
  theme(panel.grid.minor = element_blank(),
        panel.grid = element_blank(),
        legend.key = element_rect(fill = NA, color = "grey"),
        legend.background = element_blank()
  )


pc<-gl.pcoa(gl.na, nfactors=5) 
pcoa_plot1 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=1, yaxis=2)

p1 <- pcoa_plot1 + my_theme +
          scale_colour_manual(values = my_colours, name = " ") +
          geom_point(size = 4)

pcoa_plot2 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=3, yaxis=4)

p2 <- pcoa_plot2 + my_theme +
  scale_colour_manual(values = my_colours, name = " ") +
  geom_point(size = 4) 


```

```{r pcoa_rup, echo=F, warning=FALSE, cache=T, message=F, fig.height=7}

ggpubr::ggarrange(p1, p2, nrow = 2, align = "hv", labels = c("A", "B"), font.label = list(size = 18, face = "plain"), label.x = 0.075, label.y = 0.98)


```


```{r lda_rup}

my_colours <- c("#77CE61", #dodii
                "#1D72F5", #E. rup
                "#DF0101", #Fernkloof A
                "#FFA5B0", #Fernkloof B FF6F5E
                "#027368", #leafy
                "#FF9326", #restiod
                "#851D41", #rup
                 
                "#F2B950", #scab
                "#60D5FD", #set
                "#9471B4",
                "dark grey") #uni

my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1)
    )

trait_data <- read.csv(here("Data", "trait_dataV3.csv"), sep = ",")
trait_data$Collection <- as.character(trait_data$Collection)

trait_data <- trait_data %>% 
    filter(Species == "E_rupestris_rupestris" | 
               Species == "E_rupestris_dodii" |
               Species == "E_rupestris_tricostata" |
               Species == "E_setacea_setacea" |
               Species == "E_setacea_scabra" |
               Species == "E_setacea_uniflora" |
               Species == "E_setacea_disticha"
    )%>% 
    droplevels() %>% 
    mutate(Clade = 
               case_when(Collection %in% c("1589", "1597", 
                                           "1599", "1593", "1598") ~ "Fernkloof A",
                         Collection %in% c("1600", "1595") ~ "Fernkloof B",
                         Collection %in% c("1614", "1610", "1569", "1568", 
                                           "1587", "1557", "1558", "1629", 
                                           "1633", "1617", "1577", "1564",
                                           "1580", "1582") ~ "Setacea",
                         Collection %in% c("1579", "1661", "1603", "1594") ~ "Restiod tricost.",
                         Collection %in% c("1639", "1573",
                                           "1578", "1565", "1602", "1616") ~ "Dodii",
                         Collection %in% c("1618", "1619", 
                                           "1621", "1648", "1632", "1586") ~ "Leafy tricost.",
                         Collection %in% c("1634", "1559", "123",  "1605", "1563") ~ "Rupestris s.s.",
                         Collection %in% c("1644", "1636", "1631", "1642", "1640", "1641") ~ "Scabra",
                         Collection %in% c("2", "1655") ~ "East. rupestris",
                         Collection %in% c("1572", "1604") ~"Uniflora",
                         Collection %in% c("1657") ~ "Wemmershoek"
               ))



trait_data <- trait_data %>% 
    mutate(inflor_position = Plant_height/Veg_height,
           relative_sheath_length = Sheath_length/Internode_dist,
           relative_leaf_length = Leaf_length/Sheath_length,
           glume_enclosure = Glume_length_outer/Spikelet_length,
           glume_ratio = Glume_length_outer/Glume_length_inner,
           spikelet_shape = Spikelet_length/Spikelet_width,
           lemma_ratio = Lemma_outer_length/Lemma_inner_length) 
trait_data <- trait_data[, c(1:8, 11:22, 25:32)]
trait_data <- trait_data[, -c(4,8,14,15,16,18,19)]


#LDA

scale_rup <- data.frame(Clade = as.character(trait_data$Clade), scale(trait_data[, c(4:12, 15:21)]))

lda_rup <- lda(scale_rup$Clade ~ ., data = scale_rup)

lda.data <- cbind(scale_rup, predict(lda_rup)$x)

ld1 <- ggplot(lda.data, aes(LD1, LD2)) +
    geom_point(aes(color = Clade), size = 2) +
    scale_colour_manual(values = my_colours) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))
ld2 <- ggplot(lda.data, aes(LD3, LD4)) +
    geom_point(aes(color = Clade), size = 2) +
    scale_colour_manual(values = my_colours) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))

#remove restiod and uni for clarity

my_colours2 <- c("#77CE61", #dodii
                "#1D72F5", #E. rup
                "#DF0101", #Fernkloof A
                "#FFA5B0", #Fernkloof B
                "#027368", #leafy
                #"#FF9326", #restiod
                "#CC1577", #rup
                 
                "#F2B950", #scab
                "#60D5FD", #set
                "dark grey") #uni
scale_rup2 <- scale_rup %>% 
    filter(!Clade == "Restiod tricost." & !Clade == "Uniflora") %>% droplevels()

lda_rup <- lda(scale_rup2$Clade ~ ., data = scale_rup2)

lda.data <- cbind(scale_rup2, predict(lda_rup)$x)

ld3 <- ggplot(lda.data, aes(LD1, LD2)) +
    geom_point(aes(color = Clade), size = 2) +
    scale_colour_manual(values = my_colours2) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))

ld4 <- ggplot(lda.data, aes(LD3, LD4)) +
    geom_point(aes(color = Clade), size = 2) +
    scale_colour_manual(values = my_colours2) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))

ggpubr::ggarrange(ld1, ld2, ld3, ld4, ncol = 2, nrow = 2, align = "hv", labels = c("A", "B", "C", "D"), font.label = list(size = 18, face = "plain"), label.x = 0.075, label.y = 0.98, common.legend = T, legend = "right")

```


```{r rup_boxplots, fig.width=8, fig.height=10.5}

my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank()
    )

#
##a <- univariate_boxplot_trait_data("Clade", "Plant_height") #
#b <- univariate_boxplot_trait_data("Clade", "Veg_height")
#c <- univariate_boxplot_trait_data("Clade", "Leaf_width")
#d <- univariate_boxplot_trait_data("Clade", "Leaf_length")
##e <- univariate_boxplot_trait_data("Clade", "Sheath_length")
#f <- univariate_boxplot_trait_data("Clade", "Internode_dist")
#g <- univariate_boxplot_trait_data("Clade", "Culm_diam")
#h <- univariate_boxplot_trait_data("Clade", "Inflor_length")
#j <- univariate_boxplot_trait_data("Clade", "Spikelet_no")
##k <- univariate_boxplot_trait_data("Clade", "pedicel_length_epu")
##l <- univariate_boxplot_trait_data("Clade", "Glume_length_outer")
##m <- univariate_boxplot_trait_data("Clade", "Glume_length_inner")
##ma <- univariate_boxplot_trait_data("Clade", "Lemma_inner_length")
##mb <- univariate_boxplot_trait_data("Clade", "Lemma_outer_length")
#n <- univariate_boxplot_trait_data("Clade", "Spikelet_length") + theme(axis.text.x = #element_text(angle = 40, vjust = 0.9, hjust = 1))
##z <- univariate_boxplot_trait_data("Clade", "Spikelet_width") +
#  #theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
##q <- univariate_boxplot_trait_data("Clade", "Glume_ratio") +
# # theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
#
#ggpubr::ggarrange(a,b, NULL,c,d,e,f,g,h,j,k,l,m,ma,mb,n,z,q, 
#                  ncol = 3, nrow = 6, 
#                  align = 'v',
#                  heights = c(rep(1,5),1.6))


a <- univariate_boxplot_trait_data("Clade", "inflor_position")
b <- univariate_boxplot_trait_data("Clade", "Veg_height")
c <- univariate_boxplot_trait_data("Clade", "Leaf_width")
d <- univariate_boxplot_trait_data("Clade", "Leaf_length")
#e <- univariate_boxplot_trait_data("Clade", "Sheath_length")
f <- univariate_boxplot_trait_data("Clade", "Internode_dist")
e <- univariate_boxplot_trait_data("Clade", "relative_sheath_length")
ea <- univariate_boxplot_trait_data("Clade", "relative_leaf_length")
g <- univariate_boxplot_trait_data("Clade", "Culm_diam")
h <- univariate_boxplot_trait_data("Clade", "Inflor_length")
j <- univariate_boxplot_trait_data("Clade", "Spikelet_no")
k <- univariate_boxplot_trait_data("Clade", "Pedicel_length_epu")
n <- univariate_boxplot_trait_data("Clade", "Spikelet_length") 
#l <- univariate_boxplot_trait_data("Clade", "Glume_length_outer")
l <- univariate_boxplot_trait_data("Clade", "glume_enclosure")
m <- univariate_boxplot_trait_data("Clade", "glume_ratio")
#m <- univariate_boxplot_trait_data("Clade", "Glume_length_inner")
#ma <- univariate_boxplot_trait_data("Clade", "Lemma_inner_length")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
ma <- univariate_boxplot_trait_data("Clade", "spikelet_shape")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
mc <- univariate_boxplot_trait_data("Clade", "Spikelet_no")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
#mb <- univariate_boxplot_trait_data("Clade", "Lemma_outer_length")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
mb <- univariate_boxplot_trait_data("Clade", "lemma_ratio")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))


#z <- univariate_boxplot_trait_data("Clade", "Spikelet_width") +
  #theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))

ggpubr::ggarrange(b,NULL,NULL,c,d, a,e,ea, f,g,h,k,l,m,n,mc, mb,ma,q, 
                  ncol = 3, nrow = 6, 
                  align = 'v',
                  heights = c(1,1,1,1,1,1.4))

```