---
title: "Phylogenetic results"
author: "Lara Wootton"
date: "11/05/2020"
output: 
  pdf_document:
    fig_caption: yes
---

```{r setup, include=F}
knitr::opts_chunk$set(cache = T,
                      echo=FALSE, warning=FALSE, message=FALSE, results = 'hide')

if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr, maptools, sp, rgdal, plotly, rgeos, ggmap, scatterpie, tidyverse, here)


```

```{r location, include=FALSE, cache=T}
#Map data
location_data <- read.csv(here("Data", "Species_metadata.csv"))

#Change to decimal degrees

Edeg <- 0
for(i in 1:length(location_data$Species_current)){
    Edeg[i] <- location_data$easting[i] + (as.numeric(location_data$e_mins[i])/60)
}


Sdeg <- 0
for(i in 1:length(location_data$Species)){
    Sdeg[i] <- location_data$southing[i] - (as.numeric(location_data$s_mins[i])/60)
}

location_data <- cbind(location_data, Sdeg, Edeg)

map_test <- get_map(location = c(17.8, -35, 24.7, -32.5 ), maptype = "terrain-background", source = "stamen", zoom = 11)

map_res <- ggmap(map_test)


```

# Results

## Sequence data

Targeted-enrichment sequencing yielded 353 loci across 108 accessions, to which _O. sativa_ was added as an outgroup. Following read trimming, an average of 228 907 reads per sample were mapped to the target loci. Of the 353 loci, an average of 200 loci per sample had reads mapped to at least 75% of their length. Phylogenetic trees inferred from the 90% coverage, and full coverage datasets were very similar, both topologically and in terms of their branch lengths, whether they were based on exons, introns or supercontigs. Therefore, only results obtained from the 90% coverage supercontig dataset are presented here. This dataset contained 307 loci, each represented in a minimum of 98 of the total 109 accessions. The concatenated alignment used for ML analysis was 698 832 bp long and contained 12.5% gap characters. Of these sites, 388 704 (55.6%) were variable, and 226 897 (32.5%) were potentially parsimony informative. The percent GC content was 39.4%. Partitionfinder reduced the number of partitions from 307 to 84.

While the majority of GBS libraries were sequenced successfully with good data yields, a single library of an _E. setacea_ subsp. _disticha_ accession (1589) failed quality control and was not sequenced. The full GBS dataset was split into three clades for analysis. The Setacea clade, represented by 95 individuals and 24 populations, originally consisted of 572 345 variants, but this decreased to 3753 variants after filtering. The Rehmannii and Ramosa clades were each represented by 48 individuals and 12 populations. Following filtering, the number of variants in the Rehmannii clade decreased from 450 635 to 3544, and in the Ramosa clade from 443 722  to 3430 variants.

## Major clade relationships

Maximum-likelihood and ASTRAL yielded well-supported, topologically-similar trees (Fig. 1, Fig. 2). In both trees, six major clades, (here termed the Australian, Rehmannii, Ramosa, Setacea, Dura and Lowland clades), are resolved with strong support, the relationships of these clades also being consistently well supported. The branches subtending these major clades are long, with branch length in the ML tree reflecting sequence divergence and that in the ASTRAL tree reflecting number of coalescent units (generations/effective population size). In the Setacea, Ramosa and Rehmannii clades, internal branches are short relative to both the stem and terminal branches, resulting in a “broom and handle” morphology (@Crisp2009) (Fig. \@ref(fig:MLtree)). In contrast, the Lowland clade has generally longer branches, with branch lengths being more evenly distributed throughout the clade. 

The Australian clade, comprising *M. stipoides* and *T. laevis* (BS = 100, LPP = 1.00), is consistently resolved as sister to the predominantly-montane Setacea clade (BS = 100, LPP = 1.00) with strong support (BS = 100, LPP = 1.00), this pair being sister to the rest of *Ehrharta* (Fig. \@ref(fig:MLtree), Fig. \@ref(fig:astraltree)). The Setacea clade itself is composed of two species, *E. setacea* and *E. rupestris*, neither of which appear be monophyletic. 

The montane Dura clade (BS = 100, LPP = 1.00), comprising *E. dura* and *E. microlaena*, is resolved with strong support (BS = 100, LPP = 1.00) as sister to a well-supported clade (BS = 100, LPP = 1.00) comprising the Rehmannii, Ramosa and Lowland clades. Within this clade, the Rehmannii (BS = 100, LPP = 1.00) and Ramosa clades (BS = 100, LPP = 1.00) are resolved as sisters with strong support (BS = 100, LPP = 1.00). The monophyly of the Lowland clade, which comprises species occurring predominately at low to mid elevations in the GCFR, is also well supported (BS = 100, LPP = 1.00), with the two analyses resolving the internal relationships of this clade identically and with high support (BS = 100, LPP = 1.00) on most nodes (Fig. \@ref(fig:MLtree), Fig. \@ref(fig:astraltree)). Relationships within the Lowlands clade are largely consistent with those reported by @Verboom2003.



## Within-clade relationships and species delimitation

### Rehmannii clade

The ML and ASTRAL analyses resolve three groups within the Rehmannii clade (Filiformis, Subspicata and Rehmannii groups) which correspond broadly to the existing subspecies of *E. rehmannii*, namely subsp. *filiformis*, subsp. *subspicata*, and subsp. *rehmannii* (Fig. \ref(fig:rehmannii_phylogeny). Where the ML analysis identifies all three groups as monophyletic with moderate to strong support (BS = 99, 77, 95, respectively), with the the Filiformis and Subspicata subgroups being sister lineages (BS = 97), the ASTRAL analysis identifies only the Filiformis and Subspicata groups as monophyletic (LPP = 0.83, 0.96, respectively), the Rehmannii group being paraphyletic (LPP = 0.97, 0.85). A sample collected near Agulhas, initially identified as *E. rehmannii* subsp. *filiformis* (1658), is included in the Subspicata clade by both analyses.

Both the sNMF analysis with K = 3 and the PCoA based on GBS-based SNP data revealed three main population clusters, which correspond to the three subclades/grades resolved by the phylogenetic analysis (Fig. \@ref(fig:rehmannii_gbs) A, Fig. \@ref(fig:reh_pcoa)). All three clusters are dominated by a different ancestral gene pool and, although there is some evidence of potential admixture, this is typically insufficient to erode the inference of three genetically-distinct clusters. For example, while Filiformis shows evidence of potential admixture with Subspicata on the Cape Peninsula (LW1; (Fig. @ref(fig:map_reh))) and Subspicata shows evidence of potential admixture with Filiformis at Pringle Bay (1590), this is limited in scale and insufficient to erase the dominant genetic affinities of these populations. A Subspicata population (1600) from the Agulhas Plain is, however, slightly anomalous in reflecting the Filiformis and Subspicata gene pools in approximately equal measure. Although accession 1658 was originally identified as *E. rehmannii* subsp. *filiformis*, both the phylogenetic analyses and the sNMF include this accession in the Subspicata group.

The cross-entrophy criterion identified K = 10 as optimal for the Rehmannii clade (Fig. Fig. \@ref(fig:rehmannii_gbs) B). At this value of K there is strong population-level genetic differentiation, with almost every locality representing a distinct ancestral gene pool. This is particularly apparent in the Filiformis cluster. Surprisingly, within the Subspicata cluster, Agulhas population 1659 shares a gene pool with Cape Peninsula population LW3, but not with the other Agulhas populations (1658, 1659, 1660, Fig. \@ref(fig:map_reh)). Of the three clades, Rehmannii shows the greatest amount of internal admixture, with the inference of two gene pools that show strong evidence of admixture at Nature’s Valley (1652). The major mode genetic structure presented for K = 10, however, reflects only 39% of the sNMF replicates.

A LDA of the morphological data separates the Rehmannii clade into three fairly discrete clusters corresponding to the putative species groups, with the exception of a few Rehmannii group individuals which were misclassified as Subspicata (Fig. \@ref(fig:reh_lda)). The Subspicata accession initially identified as *E. rehmannii* subsp. *filiformis* (1658) clusters with the other Subspicata accessions. Univariate comparisons reveal that Rehmannii plants are generally taller than those of Subspicata, their inflorescences having more spikelets and protrude higher above the leaves (Fig. \@ref(fig:reh_boxplots)). Conversely, Filiformis plants are more delicate than either Rehmannii or Subspicata, with finer leaves, finer culms, and smaller and fewer spikelets (Fig. \@ref(fig:reh_boxplots)).



### Ramosa clade

In contrast to the existing taxonomy, which describes two subspecies within *E. ramosa*, both the ML and ASTRAL trees resolve three principal subclades within the Ramosa clade (Fig.  \@ref(fig:ram_minor_clades)). The composition of these subclades, however, differs subtly between the two analyses. The Eastern Ramosa group (from the Swartberg and Langeberg) is monophyletic within the ASTRAL tree (LPP = 0.88), but consists of a grade in the ML tree. The converse is true for the Western ramosa group (Boland Mountains to Cederberg), which is monophyletic in the ML tree (BS = 83), but paraphyletic in the ASTRAL tree (LPP = 0.71).  In contrast, the Aphylla group (Cape Peninsula to Betty's Bay) is monophyletic with strong support in both trees (BS = 100, LPP = 0.95). The _E. ramosa_ subsp. _aphylla_ and _E. ramosa_ subsp. _ramosa_ specimens sampled from sites of sympatry at Milner Peak and Limietberg are not recovered as sisters in either tree, potentially indicating the presence of multiple taxa within the Western Ramosa clade.

The admixture results showed limited support for three ancestral populations (K = 3) (Fig.  \@ref(fig:ram_gbs) A) that correspond to the recovered phylogenetic groups. While all Aphylla individuals were assigned to the same ancestral population with high probability, most Western Ramosa individuals shared a large proportion of genomic information with the Aphylla group. A population from Milner Peak (1624) is exceptional, as all individuals from this locality were assigned to a genetic cluster largely distinct to that of Aphylla. Although the dominant gene pool in the Aphylla cluster (blue in Fig. \@ref(fig:ram_gbs) also occurs in the Eastern Ramosa cluster, a high frequency of a third gene pool (red in Fig. \@ref(fig:ram_gbs) distinguishes the latter.

The cross-entrophy criterion identifies K = 8 (Fig. \@ref(fig:ram_ce)) as optimal for the Ramosa clade. At this level, the Aphylla group is dominated by a single ancestral gene pool that is largely absent from the Western and Eastern Ramosa groups, thereby corroborating the phylogenetic distinctness of Aphylla (Fig.  \@ref(fig:ram_gbs) B). In contrast, the Western and Eastern Ramosa clades both show strong structure at K = 8, with most populations representing a distinct ancestral gene pool. The exceptions are the Milner Peak (1622) and Limietberg (1609) populations of the Western Ramosa group, which share an ancestral gene pool, and are resolved as sister taxa in the phylogenetic trees (Fig.  \@ref(fig:ram_gbs) B, Fig. \@ref(fig:ram_minor_clades)), as well as the Swartberg (1656) and Seweweekspoort (1654) populations of the Eastern Ramosa group, which again are resolved as sisters in the phylogenetic tree (Fig.  \@ref(fig:ram_gbs) B,  Fig. \@ref(fig:ram_minor_clades)). Interestingly, the sympatric Milner Peak populations (1622 and 1624) do not share genetic information, despite being collected within a few kilometres of each other. The strong genetic structure of the Western Ramosa group is also evident in the PCoA (Fig.  \@ref(fig:ram_pcoa)), especially in the PC3 vs PC4 biplot. By contrast, the Eastern Ramosa cluster appears much more coherent in the PCoA.

An LDA based on trait data confirms the morphological distinctness of the three Ramosa groups, although there is some overlap between the Western and Eastern Ramosa groups. Consistent with the genetic data, the Western Ramosa group tends to show greater variability in most traits, often encompassing much of the trait range of the other two groups (Figure  \@ref(fig:ram_boxplots)). In general, Aphylla group plants tend to have longer leaves, and longer glumes relative to the spikelets, than plants in the Eastern and Western Ramosa groups. On the other hand, Eastern Ramosa plants have generally longer inflorescences and spikelets which are longer relative to their width.

### Setacea clade

The phylogeny of the Setacea clade is characterised by extremely short internal branch lengths which are often poorly supported (Fig. \@ref(fig:rup_minor_clades)). Nonetheless, it is clear that both _E. setacea_ and _E. rupestris_, as currently defined, are polyphyletic.  Four major clades are recovered with good support by the ML and ASTRAL analyses. These are: (i) the Eastern Rupestris clade (BS = 100, LPP = 1.0); (ii) the Western Rupestris clade (BS = 100, LPP = 1.0); (iii) a clade comprising the Fernkloof, Uniflora, and Restioid Tricostata clades (BS = 99, LPP = 0.74), with Uniflora and Restioid tricostata recovered as sisters in both trees (BS = 100, LPP = 0.72); and (iv) a clade consisting of the Setacea, Dodii, Leafy Tricostata and Scabra clades (BS = 97, LPP = 0.68). There is some incongruence at this level, however, with the ML analysis embedding the Western Rupestris clade within clade (iv), and so rendering the latter paraphyletic.

Within clade (iii), the monophyly of both the Uniflora clade (BS = 100, LPP = 1.00), composed of all accessions of *E*. *setacea* subsp. *uniflora*, and the Restioid Tricostata clade (BS = 100, LPP = 0.97), comprising *E. rupestris* subsp. *tricostata* accessions from the Cape Peninsula, Hottentots Holland to Fernkloof area, are well supported, and these clades are retrieved as sister with good support (BS = 100, LPP = 0.72, Fig. \@ref(fig:rup_minor_clades)). However, where the ML analysis retrieves a weakly supported (BS = 59) Fernkloof clade, comprising *E. setacea* subsp. *setacea* and *E. setacea* subsp. *disticha* plants from the Fernkloof Reserve in Hermanus and *E. setacea* subsp. *disticha* plants from the Buffelstalberg near Betty’s Bay, this assemblage is paraphyletic (LPP = 0.37) in the ASTRAL tree. Both trees, however, reflect good support for the monophyly of a clade comprising two Fernkloof accessions of *E. setacea* subsp. *setacea* (1595 and 1600) sampled from boggy depressions (BS = 100; LPP = 1.00) and a clade consisting of two Fernkloof accessions of *E. setacea* subsp. *disticha* (1597 and 1598) plus two Fernkloof accessions of *E. setacea* subsp. *setacea* (1593 and 1599) sampled from dry, rocky slopes (BS = 98, LPP = 0.88).

 Although clade (iv), which encompasses most accessions of *E. setacea* subsp. *setacea*, *E. setacea* subsp. *scabra, E. rupestris* subsp. *dodii* and eastern accessions of *E. rupestris* subsp. *tricostata*, is monophyletic in the ASTRAL tree, it is paraphyletic in the ML tree owing to the inclusion of the Western Rupestris clade (Fig. \@ref(fig:rup_minor_clades)). Within clade (iv), there are high levels of incongruence due to poorly supported internal relationships. However, both trees reflect good support for the monophyly of the Scabra clade, comprising all accessions of *E. setacea* subsp. *scabra* (BS = 100, LPP = 1.00), and the Dodii clade, comprising all accessions of *E. rupestris* subsp. *dodii* from the western GCFR (BS = 96, LPP = 0.95). In addition, the ASTRAL tree resolves both a monophyletic Leafy Tricostata clade, comprising inland *E. rupestris* subsp. *tricostata* accessions from Ceres to Robinson's Pass (LPP = 0.70), and a monophyletic Setacea clade, comprising all non-Fernkloof accessions of *E. setacea* subsp. *setacea*, albeit with weak support (LPP = 0.56). However, the ML analysis retrieves neither of these clades, with both assemblages being paraphyletic in the ML tree. Finally, two accessions, originally designated as *E. rupestris* subsp. *tricostata* (1640 and 1657) occupy isolated positions within clade (iv). Although accession 1640 is retrieved as sister to the Scabra clade in both trees with good support (BS = 100, LPP = 1.00), the position of accession 1657 in the ASTRAL tree is uncertain, while in the ML tree it is embedded in the Setacea grade. 

The sNMF analysis with K = 9 largely corroborates the genetic coherence of the nine named clades revealed by the phylogenetic analyses. Although the Dodii, Leafy Tricostata, Restioid Tricostata and Setacea clades show some evidence of between-group admixture, the genomic composition of most clades is highly distinct (Fig. \@ref(fig:rup_gbs) A). This is also apparent from the PCoA, in which each group forms a well-defined cluster (Fig. \@ref(fig:rup_pcoa)). The exceptions to this pattern are the Fernkloof (1589 and 1597) and Scabra clades, which share a very similar and highly admixed genomic profile under K = 9, although the PCoA resolves these clades as distinct (Fig.). Beyond the nine named clades, the sNMF with K = 9 assigns the Wemmershoek population represented by accession 1657 to a unique ancestral population. In addition, within the Fernkloof clade, the genomic profile of accession 1600 (*E. setacea* subsp. *setacea* from Fernkloof) is distinct from that of the other Fernkloof clade accessions included in the analysis. 

The cross-entrophy criterion identifies K = 19 as optimal within the Setacea clade. Under K = 19, almost every clade shows substructure, although there is evidence of within-group admixture in the Setacea, Restioid Tricostata, Leafy Tricostata, Dodii and Eastern Rupestris clades (Fig. \@ref(fig:rup_gbs) B). Within the Fernkloof clade, the two *E. setacea* subsp. *disticha* (1589 and 1597) populations share some genomic information with each other, but not with 1600. Based on the results of the phylogenetic and sNMF analyses, the Fernkloof group was split into two for the morphological analysis, with Fernkloof A consisting of accessions assigned to *E. setacea* subsp. *disticha* (1589 and 1597) and accessions from dry, rocky slopes assigned to *E. setacea* subsp. *setacea* (1593 and 1599), and Fernkloof B containing the accessions from flat, boggy sites assigned to *E. setacea* subsp. *setacea* (1595 and 1600).

A LDA of the morphological data shows that both the Uniflora and Restioid Tricostata groups are morphologically divergent from the remainder of the Setacea clade (Fig. \@ref(fig:rup_lda) A). Where Restioid Tricostata plants are characterised by long culm internodes which are barely covered by the leaf sheaths, short pedicels and a small glume to spikelet length ratio, Uniflora plants are tall and gracile, having exceptionally thin culms and fine leaves. In addition, the inflorescences of Uniflora plants almost always contain two spikelets, in which the glumes considerably exceed the florets (Fig. \@ref(fig:rup_boxplots)). The LDA, particularly with the Restioid Tricostata and Uniflora groups omitted, also reveals clear pairwise separation between most groups at least along one of the discriminant axes (LD1-LD4; Fig. \@ref(fig:rup_lda) C & D). Exceptions to this pattern are the Dodii and Leafy Tricostata groups, which show considerable overlap on all discriminant axes, and potentially the Eastern Rupestris and Western Rupestris groups which, despite being morphologically close, show minimal overlap (Fig. \@ref(fig:rup_lda) C & D). Despite their lack of separation in multivariate space, the Dodii and Leafy Tricostata groups can be clearly differentiated on the basis of spikelet number per inflorescence (Fig. \@ref(fig:rup_boxplots)). Likewise, Western Rupestris plants are distinguishable from Eastern Rupestris plants on the basis of their more robust character, having thicker culms, generally larger leaves (leaf length and width), and relatively more elongate spikelets (spikelet length:width ratio; Fig. \@ref(fig:rup_boxplots)). Significantly, although both contain accessions initially assigned to *E. setacea* subsp. *setacea*, the LDA classifies the Fernkloof A and Fernkloof B groups into different clusters, these groups being distinguishable primarily on the basis of floral characteristics, such as the position of the inflorescence above the leaves, spikelet length and the ratio of lemma lengths ( Fig. @ref(fig:rup_boxplots)). In addition, the LDA identifies both these groups as being distinct from the Setacea group, Setacea plants being distinguishable from Fernkloof A plants on the basis of inflorescence length and from Fernkloof B plants on the basis of shorter spikelets (Fig. \@ref(fig:rup_lda), Fig. @ref(fig:rup_boxplots)).




```{r MLtree, echo=F, fig.height=11.5, fig.width=9, warning=F, fig.cap="Maximum-likelihood tree constructed using IQTREE with Partitionfinder from 307 concatenated supercontig loci. Red numbers show ultrafast bootstraps (BS). Branch lengths represent number of substitutions. The outgroup branch length, coloured grey, has been shortened by a factor of 2.5 for clarity."}
#ML tree

tree <- read.tree(here("Data","boot_relaxed_clustering2_ninety_coverage_supercontig_21Apr.treefile"))
tree <- root(tree, outgroup = "Oryza_sativa", resolve.root = T) #Re-root

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree$tip.label[tree$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Species[[i]],
        paste("(",
        species_names$SampleID[[i]],
        ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

tree$tip.label[tree$tip.label == "Tetrarrhena laevis () "] <- "Tetrarrhena laevis"
tree$tip.label[tree$tip.label == "Microlaena stipoides () "] <- "Microlaena stipoides"
tree$tip.label[tree$tip.label == "Oryza_sativa"] <- "Oryza sativa"


#For visualisation purposes, cut the outgroup branch length by half
tree$edge.length[length(tree$edge.length)] <- tree$edge.length[length(tree$edge.length)]/2

gg_tree <- ggtree(tree, right = T)
d <- gg_tree$data
d <- d[!d$isTip,]
d <- d[!d$label=="Root",]
d <- separate(d, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)
da <- d[d$node %in% c(167, 119, 217, 118,117, 120,168,170,189,171,113),]

low <- gg_tree$data
lowl <- low[low$node %in% c(111,203,204,205,206,207,208,
                            209,210,211,212,213,214,215,
                            216,114,115,116,112),]
lowl <- separate(lowl, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)
lowl$bootstrap[lowl$node == 111] <- "100"

tree_group <- groupClade(tree, .node = 217)

ggtree(tree_group, aes(colour = group), right = T) +
    scale_colour_manual(values = c("grey", "black")) +
    geom_tiplab(size = 2.4, colour = "black") +
    geom_treescale(x = -0.001, offset = 0.6, linesize = 0.6) +
    geom_nodelab(data = da, aes(label = bootstrap),
                 size = 2.5,
                 colour = "red",
                 nudge_x = -0.0033,
                 nudge_y = -0.69) +
    geom_nodelab(data = lowl, aes(label = bootstrap),
                 size = 2.5,
                 colour = "red",
                 nudge_x = -0.0033,
                 nudge_y = -0.69)+
    geom_rootedge(rootedge = 0.01)+
    
    #Major clades
    
    geom_cladelabel(node = 167, 
                    label = "Australian clade",
                    offset = 0.025,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.5) +
    geom_cladelabel(node = 113, 
                    label = "Lowlands clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.2,
                    offset = 0.04,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001) +
    geom_cladelabel(node = 189, 
                    label = "Rehmannii clade",
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.2,
                    offset = 0.065,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001) +
    geom_cladelabel(node = 171, 
                    label = "Ramosa clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.2,
                    offset = 0.06231,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001
    ) +
    geom_cladelabel(node = 120, 
                    label = "Setacea clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.3,
                    offset = 0.07,
                    #angle = 90,
                   # hjust = 0.5,
                    offset.text = 0.001) +
    geom_cladelabel(node = 168, 
                    label = "Dura clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.3,
                    offset = 0.032,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001) +
    geom_cladelabel(node = 116, 
                    label = "Dummy clade", 
                    #align = T,
                    colour = "white",
                    fontsize = 3,
                    extend = 0.3,
                    offset = 0.14,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.001) 


```


```{r astraltree, echo=F, fig.height=11.5, fig.width=9, warning=F, fig.cap="ASTRAL tree constructed using IQTREE gene trees from 307 supercontig loci. Branch lengths represent coalescent units. Tip branches are an arbitrary length not calculated by ASTRAL. Red numbers show local posterior probabilities (LPP)."}

if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr)

tree <- read.tree(here("Data", "astral_ninety_coverage_supercontig7Apr.tre"))

tree$edge.length[is.na(tree$edge.length)] <- 0.35 #tip branches

tree <- root(tree, outgroup = "Oryza_sativa", resolve.root = T)

tree$edge.length[1] <- 0.1 #root branch

#Rename tips

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree$tip.label[tree$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Species[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

tree$tip.label[tree$tip.label == "Tetrarrhena laevis () "] <- "Tetrarrhena laevis"
tree$tip.label[tree$tip.label == "Microlaena stipoides () "] <- "Microlaena stipoides"
tree$tip.label[tree$tip.label == "Oryza_sativa"] <- "Oryza sativa"



gg_tree <- ggtree(tree, right = T)
d <- gg_tree$data
d <- d[!d$isTip,]
d <- d[!d$label=="Root",]
da <- d[d$node %in% c(127, 184, 131, 185, 199, 132, 217, 136, 135, 137, 133),]
da$label[da$label == "1"] <- "1.00"


low <- gg_tree$data
lowl <- low[low$node %in% c(128,129,130,121,
                            122,123,124,125,126,
                            111,112,113,114,115,
                            116,117,118,119,120),]

lowl$label[lowl$label == "1"] <- "1.00"
lowl$label[lowl$label == "0.6"] <- "0.60"

ggtree(tree, right = T) +
    geom_treescale(x = -0.18, offset = 0.6, linesize = 0.6) +
    geom_tiplab(size = 2.4, colour = "black") +
    geom_rootedge(rootedge = 0.3) +
    geom_nodelab(data = da, aes(label = label),
                 size = 2.5,
                 nudge_x = -0.15,
                 nudge_y = -0.72,
                 colour = "#DF0101"
    ) +
    geom_nodelab(data = lowl, aes(label = label),
                 size = 2.5,
                 nudge_x = -0.15,
                 nudge_y = -0.72,
                 colour = "#DF0101"
    ) +
    
    #Major clades
    
    geom_cladelabel(node = 136, 
                    label = "Australian clade",
                    offset = 1.4,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.5,
                    offset.text = 0.04) +
    geom_cladelabel(node = 127, 
                    label = "Lowlands clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.2,
                    offset =1.7,
                    angle = 90,
                    hjust = 0.5,
                    offset.text = 0.07) +
    geom_cladelabel(node = 185, 
                    label = "Rehmannii clade",
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.2,
                    offset = 2.8,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.04) +
    geom_cladelabel(node = 199, 
                    label = "Ramosa clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.2,
                    offset = 2.5,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.04
    ) +
    geom_cladelabel(node = 137, 
                    label = "Setacea clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.3,
                    offset = 2.6,
                    angle = 0,
                    #hjust = 0.5,
                    offset.text = 0.04) +
    geom_cladelabel(node = 133, 
                    label = "Dura clade", 
                    #align = T,
                    colour = "#1D4F9F",
                    fontsize = 4.5,
                    extend = 0.3,
                    offset = 1,
                    #angle = 90,
                    #hjust = 0.5,
                    offset.text = 0.04)

```



```{r rehmannii_phylogeny, echo=F, warning=F, fig.width=9, fig.height=11, message=F, cache=F, results='hide',fig.keep='all', fig.cap="Phylogenetic relationships of the Rehmannii clade. A) Maximum-likelihood tree constructed using IQTREE with Partitionfinder from 307 concatenated supercontig loci. Red numbers show ultrafast bootstraps (BS). Branch lengths represent number of substitutions. B) ASTRAL tree constructed using IQTREE gene trees from 307 supercontig loci. Branch lengths represent coalescent units (generations/ effective population size). Tip branches are an arbitrary length not calculated by ASTRAL III. Red numbers represent local posterior probabilities (LPP)."}


if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr)


#ML###

tree_ml <- read.tree(here("Data","boot_relaxed_clustering2_ninety_coverage_supercontig_21Apr.treefile"))
tree_ml <- root(tree_ml, outgroup = "Oryza_sativa", resolve.root = T)


#Rename tips
species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ml$tip.label[tree_ml$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

reh_clade <- extract.clade(tree_ml, 189)
reh_tree <- ggtree(reh_clade, right = T)
reh_dat <- reh_tree$data

reh_dat2 <- reh_dat[!reh_dat$isTip,]
reh_dat2 <- separate(reh_dat2, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)

ml_reh <- ggtree(reh_clade, size = 1, right = T) +
    geom_tiplab(size = 3.7, offset = 0.001) +
    geom_treescale(x = -0.002, width = 0.005, offset = 0.15, linesize = 0.6) +
    geom_rootedge(rootedge = 0.004) +
    geom_nodelab(data = reh_dat2, aes(label = bootstrap),
                 nudge_x = -0.0019,
                 nudge_y = -0.11,
                 colour = "#DF0101",
                 size = 4) +
    geom_cladelabel(node = 23, 
                    label = "Subspicata clade",
                    colour = "#F3A10E",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0025,
                    angle = -90,
                    align = T,
                    hjust = 0.5,
                    fontsize = 5) +
    geom_hilight(node=23, fill="#F3A10E", alpha = 0.15, extend = 0.034) +
    
    geom_cladelabel(node = 27, 
                    label = "Rehmannii clade",
                    colour = "#297AB1",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0025,
                    angle = -90,
                    hjust = 0.5,
                    align = T,
                    fontsize = 5) +
    geom_hilight(node=27, fill="#297AB1", alpha = 0.15, extendto = 0.0495) +
    
    geom_cladelabel(node = 18, 
                    label = "Filiformis clade",
                    colour = "#0AB276",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0025,
                    angle = -90,
                    hjust = 0.5,
                    align = T,
                    fontsize = 5) +
    geom_hilight(node=18, fill="#0AB276", alpha = 0.15, extendto = 0.0505)


#Astral ###


tree_ast <- read.tree(here("Data", "astral_ninety_coverage_supercontig7Apr.tre"))

tree_ast$edge.length[is.na(tree_ast$edge.length)] <- 0.35 #tip branches

tree_ast <- root(tree_ast, outgroup = "Oryza_sativa", resolve.root = T)

#Rename tips

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ast$tip.label[tree_ast$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

reh_ast_clade <- extract.clade(tree_ast, 185)

reh_tree_ast <- ggtree(reh_ast_clade, right = T)

reh_tree_ast <- flip(reh_tree_ast, 24,19)

reh_ast_dat <- reh_tree_ast$data
reh_ast_dat <- reh_ast_dat[!reh_ast_dat$isTip,]
reh_ast_dat$label[reh_ast_dat$label == "1"] <- "1.00"


ast_reh <- ggtree(reh_ast_clade, right = T, size = 1, ladderize = T) +
    geom_tiplab(size = 3.7, offset = 0.02) +
    geom_rootedge(rootedge = 0.05) +
    geom_treescale(x = -0.08, offset = 0.15, linesize = 0.6, width = 0.2)+
    geom_nodelab(data = reh_ast_dat, aes(label = label),
                 nudge_x = -0.11,
                 nudge_y = -0.12,
                 colour = "#DF0101",
                 size = 4) +
    geom_cladelabel(node = 26, 
                    label = "Subspicata clade",
                    colour = "#F3A10E",
                    extend = 0.2,
                    offset = 1.65,
                    offset.text = 0.1,
                    angle = -90,
                    hjust = 0.5,
                    fontsize = 5,
                    align = T) +
    geom_hilight(node=26, fill="#F3A10E", alpha = 0.15, extendto = 2.4) +
    
    geom_cladelabel(node = 17, 
                    label = "Rehmannii clade",
                    colour = "#297AB1",
                    extend = 0.55,
                    offset = 1.65,
                    offset.text = 0.1,
                    angle = -90,
                    hjust = 0.15,
                    fontsize = 5,
                    align = T) +
    geom_cladelabel(node = 25, 
                    label = "",
                    colour = "#297AB1",
                    extend = 0.55,
                    offset = 1.65,
                    offset.text = 0.07,
                    angle = -90,
                    hjust = 0,
                    fontsize = 5,
                    align = T)+
    geom_hilight(node=25, fill="#297AB1", alpha = 0.15, extendto = 2.2) +
    geom_hilight(node=17, fill="#297AB1", alpha = 0.15, extendto = 2.2) +
    
    geom_cladelabel(node = 19, 
                    label = "Filiformis clade",
                    colour = "#0AB276",
                    extend = 0.3,
                    offset = 1.65,
                    offset.text = 0.1,
                    angle = -90,
                    hjust = 0.5,
                    fontsize = 5,
                    align = T) +
    geom_hilight(node=19, fill="#0AB276", alpha = 0.15, extendto = 2.3) 

ast_reh <- flip(ast_reh, 24,19)

#require(ggpubr)
#ggarrange(ml_reh, ast_reh, labels = c("A", "B"), font.label = list(size = 19, face = "plain"), label.x = 0.07, label.y = 0.98, plotlist = F)

cowplot::plot_grid(ml_reh, ast_reh, nrow = 1)

#require(gridExtra)
#require(grid)
#arrangeGrob(test) + grid.draw(test)
```


```{r reh_ce, fig.width=4, fig.height=8.5, fig.cap="The cross-entropy criterion plotted for each value of K between K = 1 and K = 20 for A) the Rehmannii clade and B) the Ramosa clade, and for each value of K between K = 1 and K = 25 for C) the Rupestris clade. Orange points show K corresponding to number of recovered phylogenetic groups, green points show the K with the lowest cross-entropy criterion."}

means <- read.csv(here("Data", "rehmannii_ce.csv"))


my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          axis.title = element_text(size = 13)
         
    )

best <- means %>% filter(k == 10)
phylo <- means %>% filter(k == 3)
a <- ggplot() +
  geom_point(data = means, aes(x = k, y = ce), size = 3.5) +
  geom_point(data = best, aes(x = k, y = ce), colour = "#009E73", size = 3.5) +
  geom_point(data = phylo, aes(x = k, y = ce), colour = "#D55E00", size = 3.5) +
  xlab("Ancestral gene pool (K) Rehmannii clade") +
  ylab("Cross-entropy") +
  my_theme

means <- read.csv(here("Data", "ramosa_ce2.csv"))
best <- means %>% filter(k == 8)
phylo <- means %>% filter(k == 3)
b <- ggplot() +
  geom_point(data = means, aes(x = k, y = ce), size = 3.5) +
  geom_point(data = best, aes(x = k, y = ce), colour = "#009E73", size = 3.5) +
  geom_point(data = phylo, aes(x = k, y = ce), colour = "#D55E00", size = 3.5) +
  xlab("Ancestral gene pool (K) Ramosa clade") +
  ylab("Cross-entropy") +
  my_theme

means <- read.csv(here("Data", "rupestris_ce.csv"))
best <- means %>% filter(k == 19)
phylo <- means %>% filter(k == 9)
c <- ggplot() +
  geom_point(data = means, aes(x = k, y = ce), size = 3.5) +
  geom_point(data = best, aes(x = k, y = ce), colour = "#009E73", size = 3.5) +
  geom_point(data = phylo, aes(x = k, y = ce), colour = "#D55E00", size = 3.5) +
  xlab("Ancestral gene pool (K) Rupestris clade") +
  ylab("Cross-entropy") +
  my_theme

ggpubr::ggarrange(a,b,c, ncol = 1, nrow = 3, align = "hv", labels = c("A", "B", "C"), font.label = list(size = 18, face = "plain"), label.x = 0.01, label.y = )
```


```{r rehmannii_gbs, echo=F, warning=FALSE, cache=T, message=F, include = T, fig.height=6.4, fig.cap='Plots of sNMF genomic assignment of individuals within the Rehmannii clade for A) K = 3, and B) K = 10. Fractions displayed at the top right of the graphs show the number of times a given sNMF repetition recovered the displayed assignments. The bars below the plots identify accession numbers and taxon groups.' }

library(pophelper)
library(gridExtra)
library(label.switching)
library(tidyr)


clump_file3 <-readQ(here("Data", "calc.clumpp.merged.K3.txt"))


#population assignments
pop_names <- read.table(here("Data", "popmap_rehmannii_reordered"), stringsAsFactors=F)

clade_names <- c(
    rep("Filiformis", 16),
    rep("Subspicata", 20),
    rep("Rehmannii", 12)
)


group_labs <- data.frame(collection = pop_names, clade = clade_names)
names(group_labs)[1] <- "collection"
group_labs <- separate(group_labs, 
                       col = collection, 
                       into = c("letter", "pop_id"),
                       sep = "A")
group_labs <- group_labs[,2:3]
group_labs$clade <- as.character(group_labs$clade)

my_colours <- c("#0AB276", #fili
                "#F3A10E", #sub
                "#297AB1" #reh
                ) 


p3 <- plotQ(clump_file3, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 4,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 3,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 3, 100/100",
            titlesize = 14,
            titlecol = "black",
            titlehjust = 1,
            titlespacer = 5,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)



#K = 10



clump_file <- readQ(here("Data", "calc.clumpp.merged.K10.txt"))

my_colours <- c(
  "#1D72F5","#DF0101","#77CE61", "#FF9326","#A945FF","#0089B2","#FDF060","#FFA6B2","#BFF217","#60D5FD"
)


p4 <- plotQ(clump_file, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 4,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 3,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 10, 39/100",
            titlesize = 14,
            titlecol = "black",
            titlehjust = 1,
            titlespacer = 5,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

ggpubr::ggarrange(arrangeGrob(p3$plot[[1]]), arrangeGrob(p4$plot[[1]]), nrow = 2,
                  labels = c("A", "B"), font.label = list(size = 19, face = "plain"), 
                  label.x = 0.001, label.y = 0.999)

```



```{r, include=F, fig.height=4}

library("vcfR")
library("adegenet")
library("dartR")
library("plotly")
library("here")

vcf <- read.vcfR(here("Data", "Final_lenient_rehmannii_28Apr.vcf"))

pop.assign<- read.csv(here("Data", "rehmannii_pop_assignment.csv"), header=T)
dat.pop.assign<-data.frame(pop.assign$Species_current, pop.assign$Pop)
colnames(dat.pop.assign)[colnames(dat.pop.assign)=="V1"] <- "pop.assign"

dat.pop.assign <- dat.pop.assign %>% 
  mutate(clade_labs =
         case_when(pop.assign.Species_current %in% c("E_rehmannii_filiformis") ~ "Filiformis", 
                   pop.assign.Species_current %in% c("E_rehmannii_rehmannii") ~ "Rehmannii",
                   pop.assign.Species_current %in% c("E_rehmannii_subspicata") ~ "Subspicata"
                   )
)

dat.pop.assign$clade_labs[dat.pop.assign$pop.assign.Pop == "A1658"] <- "Subspicata"

#convert vcf to genlight
gl <- vcfR2genlight(vcf)

##Assign pop names so that R knows how to colour the PCoA points
strata(gl)<-data.frame(dat.pop.assign)
setPop(gl)<-~clade_labs

#Before PCoA, find & exclude the SNPs with missing data:
toRemove <- is.na(glMean(gl, alleleAsUnit = FALSE)) 
gl.na <- gl[, !toRemove]


my_colours <- c("#0AB276", #fili
                "#F3A10E", #sub
                "#297AB1" #reh
) 

my_theme <- theme_bw() +
  theme(panel.grid.minor = element_blank(),
        panel.grid = element_blank(),
        legend.key = element_rect(fill = NA, color = "grey"),
        legend.background = element_blank(),
        legend.text = element_text(size = 13)
  )


pc<-gl.pcoa(gl.na, nfactors=5) 
pcoa_plot1 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=1, yaxis=2)

p1 <- pcoa_plot1 + my_theme +
          scale_colour_manual(values = my_colours, name = " ") +
          geom_point(size = 4)

pcoa_plot2 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=3, yaxis=4)

p2 <- pcoa_plot2 + my_theme +
  scale_colour_manual(values = my_colours, name = " ") +
  geom_point(size = 4) 


```

```{r pcoa_reh, echo=F, warning=FALSE, cache=F, message=F, fig.height=7, fig.width=5, fig.cap='Representation of genetic similarity between individuals in the Rehmannii clade using a principle coordinates analysis of SNP variants. A) Principal component axes 1 and 2, B) principal component axes 3 and 4.' }

ggpubr::ggarrange(p1, p2, nrow = 2, align = "hv", labels = c("A", "B"), font.label = list(size = 18, face = "plain"), label.x = 0.12, label.y = 0.98, common.legend = T, legend = "right")



```


```{r map_reh, fig.width=8, fig.height=8, fig.cap='Map showing average sNMF assignments per population at K = 9 for the Setacea clade.  Colours correspond to those of Figure 5A.'}
clump_file3 <- readQ(here("Data", "calc.clumpp.merged.K3.txt"))

gen_dat <- as.data.frame(clump_file3$`calc.clumpp.merged.K3-1`)

pop_names <- read.table(here("Data", "popmap_rehmannii_reordered"), stringsAsFactors=F)

gen_dat <- cbind(pop_names, gen_dat)

gen_dat <- separate(gen_dat, 
                    col = V1, 
                    into = c("letter", "Pop"),
                    sep = "A")
gen_dat$Pop[gen_dat$Pop == "LW1"] <- 1
gen_dat$Pop[gen_dat$Pop == "LW3"] <- 3

gen_dat$Pop <- as.numeric(gen_dat$Pop)

gen_dat_reduced <- gen_dat %>% 
  group_by(Pop) %>% 
  mutate(C1 = mean(Cluster1),
         C2 = mean(Cluster2),
         C3 = mean(Cluster3)) %>% 
  as.data.frame()

gen_dat_reduced <- gen_dat_reduced[, c(2,6:8)] 
gen_dat_reduced <- gen_dat_reduced %>% distinct()

reh_location <- location_data %>% 
  filter(Major_groups == "rehmanni") %>% 
  distinct()


combined <- left_join(gen_dat_reduced, reh_location, by = "Pop")
combined <- combined[,c(1:6, 15:16)]


combined$Sdeg[combined$Pop == 1590] <- -34.27
combined$Edeg[combined$Pop == 1590] <- 18.7532

combined$Sdeg[combined$Pop == 1591] <- -34.35
combined$Edeg[combined$Pop == 1591] <- 18.99
combined$Sdeg[combined$Pop == 3] <- -34.29
combined$Sdeg[combined$Pop == 1] <- -34.05

combined$Pop[1] <- "LW1"
combined$Pop[6] <- "LW3"

#Plot

my_colours <- c("#0AB276", #fili
                "#F3A10E", #sub
                "#297AB1" #reh
) 


map_res +
  geom_scatterpie(data = combined, 
                  aes(x = Edeg, y = Sdeg,  r = rep(0.13, 36)), 
                  cols = c("C1", "C2",  "C3"),
                  pie_scale = 5,
                  size = 0.1) +
  scale_fill_manual(values = my_colours) +
  geom_text(data = combined, 
            aes(x = Edeg, y = Sdeg),
            label = combined$Pop,
            nudge_y = -0.22,
            size = 3.2
  ) +
  theme(legend.position = "none") +
  coord_equal() +
  xlab("Longitude") +
  ylab("Latitude")

```

```{r reh_lda, fig.width =6.5, fig.height = 5, fig.cap='A linear discriminant analysis of morphological data for the Rehmannii clade. The 17 included traits are described in Table 1.'}
library(MASS)
library(dplyr)


trait_data <- read.csv(here("Data", "trait_dataV3.csv"), sep = ",")
trait_data$Collection <- as.character(trait_data$Collection)

trait_data <- trait_data %>% 
    filter(Species == "E_rehmannii_filiformis" | 
               Species == "E_rehmannii_rehmannii" |
               Species == "E_rehmannii_subspicata" ) %>% 
    droplevels() %>% 
    mutate(Clade = case_when(Collection %in% c("1649", "1652", "1653", "1647") ~ "Rehmannii",
                             Collection %in% c( "1571", "1591", "1592", 
                                                "1615", "1567", "1", 
                                                "4", "1583") ~ "Filiformis",
                             Collection %in% c("1658","1660", "1590", "1659", "3") ~ "Subspicata"
    )
    )

trait_data <- trait_data %>% 
    mutate(inflor_position = Plant_height/Veg_height,
           relative_sheath_length = Sheath_length/Internode_dist,
           relative_leaf_length = Leaf_length/Sheath_length,
           glume_enclosure = Glume_length_outer/Spikelet_length,
           glume_ratio = Glume_length_outer/Glume_length_inner,
           spikelet_shape = Spikelet_length/Spikelet_width,
           lemma_ratio = Lemma_outer_length/Lemma_inner_length)

trait_data <- trait_data[, c(1:8, 11:22, 25:32)]
trait_data <- trait_data[, -c(4,8,14,15,16,18,19)]

my_colours <- c("#0AB276", "#297AB1", "#F3A10E")

my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          axis.title = element_text(size = 12)
         
    )


scale_reh <- data.frame(Clade = as.character(trait_data$Clade), scale(trait_data[, c(4:13, 15:21)]))

lda_rehmannii <- lda(scale_reh$Clade ~ ., data = scale_reh)

lda.data <- cbind(scale_reh, predict(lda_rehmannii)$x)

ggplot(lda.data, aes(LD1, LD2)) +
    geom_point(aes(color = Clade), size = 3) +
    scale_colour_manual(values = my_colours) +
    my_theme

```

```{r reh_boxplots, fig.height=9, message=F, fig.show='first',fig.cap='Boxplots of morphological traits used in the linear discriminant analysis for the Rehmannii clade. See Table 1 for trait decriptions. Where applicable, measurements are in mm.'}
my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank()
    )

univariate_boxplot_trait_data <- function(x, y) {
    ggplot(trait_data, aes_string(x, y)) +
        geom_boxplot(fill = my_colours,
                     width = 0.6) +
        my_theme
}



a <- univariate_boxplot_trait_data("Clade", "inflor_position") + ylab("Inflorescence position")
b <- univariate_boxplot_trait_data("Clade", "Veg_height")+ ylab("Vegetative height")
c <- univariate_boxplot_trait_data("Clade", "Leaf_width")+ ylab("Leaf width")
d <- univariate_boxplot_trait_data("Clade", "Leaf_length") +ylab("Leaf length")
#e <- univariate_boxplot_trait_data("Clade", "Sheath_length")
f <- univariate_boxplot_trait_data("Clade", "Internode_dist")+ ylab("Internode distance")
e <- univariate_boxplot_trait_data("Clade", "relative_sheath_length")+ ylab("Relative sheath length")
ea <- univariate_boxplot_trait_data("Clade", "relative_leaf_length")+ ylab("Relative leaf length")
g <- univariate_boxplot_trait_data("Clade", "Culm_diam")+ ylab("Culm diameter")
h <- univariate_boxplot_trait_data("Clade", "Inflor_length")+ ylab("Inflorescence length")
#q <- univariate_boxplot_trait_data("Clade", "Spikelet_no")
k <- univariate_boxplot_trait_data("Clade", "Pedicel_length_epu")+ ylab("Pedicel length")
n <- univariate_boxplot_trait_data("Clade", "Spikelet_length") + theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1, size = 12, colour = "black"))+ ylab("Spikelet length (mm)")
#l <- univariate_boxplot_trait_data("Clade", "Glume_length_outer")
l <- univariate_boxplot_trait_data("Clade", "glume_enclosure")+ ylab("Glume enclosure")
m <- univariate_boxplot_trait_data("Clade", "glume_ratio")+ ylab("Glume ratio")
#q <- univariate_boxplot_trait_data("Clade", "Glume_length_inner")
#ma <- univariate_boxplot_trait_data("Clade", "Lemma_inner_length")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
ma <- univariate_boxplot_trait_data("Clade", "spikelet_shape")+ ylab("Spikelet shape")
mc <- univariate_boxplot_trait_data("Clade", "Spikelet_no")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1, size = 12, colour = "black"))+ ylab("Spikelet no.")
#mb <- univariate_boxplot_trait_data("Clade", "Lemma_outer_length")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
mb <- univariate_boxplot_trait_data("Clade", "lemma_ratio")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1, size = 12, colour = "black"))+ ylab("Lemma ratio")
z <- univariate_boxplot_trait_data("Clade", "Striation_no")+ ylab("Striation no.")

#z <- univariate_boxplot_trait_data("Clade", "Spikelet_width") +
  #theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))

null <- ggplot() +
  geom_point(data = means, aes(x = k, y = ce), size = 3.5, alpha = 0) + theme_void()

ggpubr::ggarrange(c,d, null, b,a,e,ea, f, g,h,k,l,z,m,ma,mc, mb,n,
                  ncol = 3, nrow = 6, 
                  align = 'v',
                  heights = c(1,1,1,1,1,1.45))

```


```{r ram_minor_clades, echo=F, warning=F, fig.width=9, fig.height=11, fig.cap="Phylogenetic relationships of Ramosa clade. A) Maximum-likelihood tree constructed using IQTREE with Partitionfinder from 307 concatenated supercontig loci. Red numbers show ultrafast bootstraps (BS). Branch lengths represent number of substitutions. B) ASTRAL tree constructed using IQTREE gene trees from supercontig 307 loci. Branch lengths represent coalescent units (generations/ effective population size). Tip branches are an arbitrary length not calculated by ASTRAL III. Red numbers represent local posterior probabilities (LPP)."}

#Ramosa

if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr)


#ML####

#Rename tips
species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ml$tip.label[tree_ml$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

ram_clade <- extract.clade(tree_ml, 171)
ram_tree <- ggtree(ram_clade, right = T)
ram_dat <- ram_tree$data

ram_dat2 <- ram_dat[!ram_dat$isTip,]
ram_dat2 <- separate(ram_dat2, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)


ml_ram <- ggtree(ram_clade, size = 1, right = T) +
    geom_tiplab(size = 3.7, offset = 0.001) +
    geom_treescale(x = -0.0022, width = 0.005, offset = 0.15, linesize = 0.6) +
    geom_rootedge(rootedge = 0.004) +
    geom_nodelab(data = ram_dat2, aes(label = bootstrap),
                 nudge_x = -0.002,
                 nudge_y = -0.17,
                 colour = "#DF0101",
                 size = 4) +
    geom_cladelabel(node = 31, 
                    label = "Aphylla clade",
                    colour = "#00A0B0",
                    extend = 0.3,
                    offset = 0.035,
                    offset.text = 0.0025,
                    angle = -90,
                    align = T,
                    hjust = 0.5,
                    fontsize = 5) +
    geom_hilight(node=31, fill="#00A0B0", alpha = 0.2, extend = 0.034) +
    geom_cladelabel(node = 23, 
                    label = "Western Ramosa clade",
                    colour = "#9BC53D",
                    extend = 0.35,
                    offset = 0.035,
                    offset.text = 0.0025,
                    angle = -90,
                    align = T,
                    hjust = 0.5,
                    fontsize = 5) +
    geom_hilight(node=23, fill="#9BC53D", alpha = 0.2, extend = 0.03) +
   
    geom_cladelabel(node = 37, 
                    label = "Eastern Ramosa clade",
                    colour = "#CC333F",
                    extend = 1.5,
                    offset = 0.035,
                    offset.text = 0.0025,
                    angle = -90,
                    align = T,
                    hjust = 0.4,
                    fontsize = 5) +
    geom_cladelabel(node = 35, 
                    label = "",
                    colour = "#CC333F",
                    extend = 0.45,
                    offset = 0.035,
                    offset.text = 0.0015,
                    angle = 90,
                    align = T,
                    hjust = 0.5,
                    fontsize = 5) +
    geom_hilight(node=36, fill="#CC333F", alpha = 0.2, extendto = 0.05) +
    geom_hilight(node=35, fill="#CC333F", alpha = 0.2, extendto = 0.05)
    

#Astral ####


#Rename tips

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ast$tip.label[tree_ast$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

ram_ast_clade <- extract.clade(tree_ast, 199)

ram_tree_ast <- ggtree(ram_ast_clade, right = T)
ram_tree_ast <- flip(ram_tree_ast,22,28)
ram_tree_ast <- flip(ram_tree_ast,23,24)

ram_ast_dat <- ram_tree_ast$data
ram_ast_dat <- ram_ast_dat[!ram_ast_dat$isTip,]
ram_ast_dat[ram_ast_dat == "1"] <- "1.00"
ram_ast_dat[ram_ast_dat == "0.7"] <- "0.70"
ram_ast_dat[ram_ast_dat == "0.4"] <- "0.40"

ast_ram <- ggtree(ram_ast_clade, right = T, size = 1, ladderize = T) +
    geom_tiplab(size = 3.7, offset = 0.02) +
    geom_rootedge(rootedge = 0.05) +
    geom_treescale(x = -0.07, width = 0.2, offset = 0.17, linesize = 0.6)+
    geom_nodelab(data = ram_ast_dat, aes(label = label),
                 nudge_x = -0.09,
                 nudge_y = -0.17,
                 colour = "#DF0101",
                 size = 4) +
    geom_cladelabel(node = 24, 
                    label = "Aphylla clade",
                    colour = "#00A0B0",
                    extend = 0.2,
                    offset = 1.2,
                    offset.text = 0.07,
                    angle = -90,
                    hjust = 0.5,
                    fontsize = 5,
                    align = T) +
    geom_hilight(node=24, fill="#00A0B0", alpha = 0.2, extendto = 1.8) +
    geom_cladelabel(node = 34, 
                    label = "Eastern Ramosa clade",
                    colour = "#CC333F",
                    extend = 0.5,
                    offset = 1.2,
                    offset.text = 0.07,
                    angle = -90,
                    hjust = 0.5,
                    fontsize = 5,
                    align = T) +
    geom_hilight(node=34, fill="#CC333F", alpha = 0.2, extendto = 1.7)+
    geom_cladelabel(node = 28, 
                    label = "Western Ramosa clade",
                    colour = "#9BC53D",
                    extend = 0.7,
                    offset = 1.2,
                    offset.text = 0.07,
                    angle = -90,
                    hjust = 0.65,
                    fontsize = 5,
                    align = T) +
    geom_cladelabel(node = 23, 
                    label = "",
                    colour = "#9BC53D",
                    extend = 0.4,
                    offset = 1.2,
                    offset.text = 0.07,
                    angle = 90,
                    hjust = 0.5,
                    fontsize = 4.5,
                    align = T) +
    geom_hilight(node=28, fill="#9BC53D", alpha = 0.2, extendto = 1.5) +
    geom_hilight(node=23, fill="#9BC53D", alpha = 0.2, extendto = 1.5)
ast_ram <- flip(ast_ram, 22, 28)
ast_ram <- flip(ast_ram,23,24)

ggpubr::ggarrange(ml_ram, ast_ram, labels = c("A", "B"), font.label = list(size = 19, face = "plain"), label.x = 0.07, label.y = 0.98)


```

```{r ram_gbs, echo=F, warning=FALSE, cache=F, message=F, include = T, fig.height=6.4, fig.cap='Plots of sNMF genomic assignment of individuals within the Ramosa clade for A) K = 3, and B) K = 8. Fractions displayed at the top right of the graphs show the number of times a given sNMF repetition recovered the displayed assignments. The bars below the plots identify accession numbers and taxon groups.'}

library(pophelper)
library(gridExtra)
library(label.switching)
library(tidyr)


clump_file <-readQ(here("Data", "calc.clumpp.merged.K3.2.txt"))

#population assignments
pop_names <- read.table(here("Data", "popmap_ramosa_reordered"), stringsAsFactors=F)

clade_names <- c(
    rep("Aphylla", 12),
    rep("Western Ramosa", 20),
    rep("Eastern Ramosa", 16)
)

group_labs <- data.frame(collection = pop_names, clade = clade_names)
names(group_labs)[1] <- "collection"
group_labs <- separate(group_labs, 
                       col = collection, 
                       into = c("letter", "pop_id"),
                       sep = "A")
group_labs <- group_labs[,2:3]
group_labs$clade <- as.character(group_labs$clade)

my_colours <- c("#00A0B0", #aphylla
                "#CC333F", #E ram
                "#9BC53D" #W ram
) 


p1 <- plotQ(clump_file, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 4,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 4,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 3, 95/100",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            titlehjust = 1,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

#K = 8


clump_file2 <- readQ(here("Data", "calc.clumpp.merged.K8.2.txt"))

my_colours <- c(
    "#1D72F5","#DF0101","#77CE61", "#FF9326","#A945FF","#0089B2","#FDF060","#FFA6B2","#BFF217","#60D5FD"
)



p2 <- plotQ(clump_file2, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 4,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 4,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 8, 67/100",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            titlehjust = 1,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

ggpubr::ggarrange(arrangeGrob(p1$plot[[1]]), arrangeGrob(p2$plot[[1]]), nrow = 2,
                  labels = c("A", "B"), font.label = list(size = 19, face = "plain"), 
                  label.x = 0.001, label.y = 0.999)
```


```{r ram_pcoa, include=F}
library("vcfR")
library("adegenet")
library("dartR")
library("plotly")

vcf <- read.vcfR(here("Data", "Ramosa_proper17June.vcf"))

pop.assign<- read.csv(here("Data", "ramosa_population_assignment.csv"), header=T)
dat.pop.assign<-data.frame(pop.assign$Species_current, pop.assign$Pop)
colnames(dat.pop.assign)[colnames(dat.pop.assign)=="V1"] <- "pop.assign"

dat.pop.assign <- dat.pop.assign %>% 
    mutate(clade_labs = c(rep("Aphylla", 8), rep("Western Ramosa", 8), 
                          rep("Aphylla", 4), rep("Western Ramosa", 12),
                          rep("Eastern Ramosa",16)) 
    )

#convert vcf to genlight
gl <- vcfR2genlight(vcf)

##Assign pop names so that R knows how to colour the PCoA points
strata(gl)<-data.frame(dat.pop.assign)
setPop(gl)<-~clade_labs


#Before PCoA, find & exclude the SNPs with missing data:
toRemove <- is.na(glMean(gl, alleleAsUnit = FALSE)) 
which(toRemove) 
gl.na <- gl[, !toRemove]

my_colours <- c("#00A0B0", #aphylla
                "#9BC53D", #W ram
                "#CC333F" #E ram
               
)

my_theme <- theme_bw() +
  theme(panel.grid.minor = element_blank(),
        panel.grid = element_blank(),
        legend.key = element_rect(fill = NA, color = "grey"),
        legend.background = element_blank(),
        legend.text = element_text(size = 13)
  )


pc<-gl.pcoa(gl.na, nfactors=5) 
pcoa_plot1 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=1, yaxis=2)

p1 <- pcoa_plot1 + my_theme +
          scale_colour_manual(values = my_colours, name = " ") +
          geom_point(size = 4)

pcoa_plot2 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=3, yaxis=4)

p2 <- pcoa_plot2 + my_theme +
  scale_colour_manual(values = my_colours, name = " ") +
  geom_point(size = 4) 


```

```{r pcoa_ram, echo=F, warning=FALSE, cache=T, message=F, fig.height=7.5, fig.width=6, fig.cap='Representation of genetic similarity between individuals in the Rehmannii clade using a principle coordinates analysis of SNP variants. A) Principal component axes 1 and 2, B) principal component axes 3 and 4.'}


ggpubr::ggarrange(p1, p2, nrow = 2, align = "hv",labels = c("A", "B"), font.label = list(size = 18, face = "plain"), label.x = 0.12, label.y = 0.98, common.legend = T, legend = "right")


```

```{r ram, cache=T}

map_test <- get_map(location = c(17.8, -35, 22.7, -32.5 ), maptype = "terrain-background", source = "stamen", zoom = 11)
map_res <- ggmap(map_test)

```

```{r ram_map, fig.width=8, fig.height=7, fig.cap='Map showing average sNMF assignments per population for the Ramosa clade at A) K = 3, B) K = 8. Colours correspond to those of Figure 11.'}
clump_file <- readQ(here("Data", "calc.clumpp.merged.K3.2.txt"))

gen_dat <- as.data.frame(clump_file$`calc.clumpp.merged.K3.2-1`)

pop_names <- read.table(here("Data", "popmap_ramosa_reordered"), stringsAsFactors=F)

gen_dat <- cbind(pop_names, gen_dat)

gen_dat <- separate(gen_dat, 
                    col = V1, 
                    into = c("letter", "Pop"),
                    sep = "A")


gen_dat$Pop <- as.numeric(gen_dat$Pop)

gen_dat_reduced <- gen_dat %>% 
    group_by(Pop) %>% 
    mutate(C1 = mean(Cluster1),
           C2 = mean(Cluster2),
           C3 = mean(Cluster3)) %>% 
    as.data.frame()

gen_dat_reduced <- gen_dat_reduced[, c(2,6:8)] 
gen_dat_reduced <- gen_dat_reduced %>% distinct()


ram_location <- location_data %>% 
    filter(Major_groups == "ramosa") %>% 
    distinct()


combined <- left_join(gen_dat_reduced, ram_location, by = "Pop")
combined <- combined[,c(1:6, 15:16)]

combined$Edeg[combined$Pop == 1624] <- 19.5790
combined$Edeg[combined$Pop == 1622] <- 19.4061


my_colours <- c("#00A0B0", #aphylla
                "#CC333F", #E ram
                "#9BC53D" #W ram
                
                
) 


m1 <- map_res +
    geom_scatterpie(data = combined, 
                    aes(x = Edeg, y = Sdeg,  r = rep(0.13, 36)), 
                    cols = c("C1", "C2",  "C3"),
                    pie_scale = 5,
                    size = 0.1) +
    scale_fill_manual(values = my_colours) +
    geom_text(data = combined, 
              aes(x = Edeg, y = Sdeg),
              label = combined$Pop,
              nudge_y = -0.22,
              size = 3.2
    ) +
    theme(legend.position = "none") +
    coord_equal() +
    xlab("Longitude") +
    ylab("Latitude")


clump_file2 <- readQ(here("Data", "calc.clumpp.merged.K8.2.txt"))

gen_dat <- as.data.frame(clump_file2$`calc.clumpp.merged.K8.2-1`)

pop_names <- read.table(here("Data", "popmap_ramosa_reordered"), stringsAsFactors=F)

gen_dat <- cbind(pop_names, gen_dat)

gen_dat <- separate(gen_dat, 
                    col = V1, 
                    into = c("letter", "Pop"),
                    sep = "A")


gen_dat$Pop <- as.numeric(gen_dat$Pop)

gen_dat_reduced <- gen_dat %>% 
    group_by(Pop) %>% 
    mutate(C1 = mean(Cluster1),
           C2 = mean(Cluster2),
           C3 = mean(Cluster3),
           C4 = mean(Cluster4),
           C5 = mean(Cluster5),
           C6 = mean(Cluster6),
           C7 = mean(Cluster7),
           C8 = mean(Cluster8)) %>% 
    as.data.frame()

gen_dat_reduced <- gen_dat_reduced[, c(2,11:18)] 
gen_dat_reduced <- gen_dat_reduced %>% distinct()


ram_location <- location_data %>% 
    filter(Major_groups == "ramosa") %>% 
    distinct()


combined2 <- left_join(gen_dat_reduced, ram_location, by = "Pop")
combined2 <- combined2[,c(1:11, 20:21)]
combined2$Edeg[combined$Pop == 1624] <- 19.5790
combined2$Edeg[combined$Pop == 1622] <- 19.4061

#Plot

my_colours <- c(
    "#1D72F5","#DF0101","#77CE61", "#FF9326","#A945FF","#0089B2","#FDF060","#FFA6B2","#BFF217","#60D5FD"
)

m2 <- map_res +
    geom_scatterpie(data = combined2, 
                    aes(x = Edeg, y = Sdeg, r = rep(0.12, 96)), 
                    cols = c("C1", "C2",  "C3", "C4", "C5", "C6",  "C7", "C8"),
                    pie_scale = 2,
                    size = 0.1) +
    scale_fill_manual(values = my_colours)+
    geom_text(data = combined2, 
              aes(x = Edeg, y = Sdeg),
              label = combined$Pop,
              nudge_y = -0.2,
              size = 3.2
    )+
    theme(legend.position = "none"
          ) +
    coord_equal() +
  xlab("Longitude")+
  ylab("Latitude")

ggpubr::ggarrange(m1,m2, ncol = 1, align = "hv", labels = c("A", "B"), font.label = list(size = 19, face = "plain"), label.x = 0.07, label.y = 0.98 )

```

```{r ram_lda, fig.width =6.5, fig.height = 5, fig.cap='A linear discriminant analysis of morphological data for the Ramosa clade. The 17 included traits are described in Table 1. '}
my_colours <- c("#00A0B0", "#DF0101",  "#9BC53D")


trait_data <- read.csv(here("Data", "trait_dataV3.csv"), sep = ",")
trait_data$Collection <- as.character(trait_data$Collection)

trait_data <- trait_data %>% 
    filter(Species == "E_ramosa_aphylla" | 
               Species == "E_ramosa_ramosa") %>% 
    droplevels() %>% 
    
    mutate(Clade = 
              case_when(Collection %in% c("1566", "1570", "1601", "1576") ~ "Aphylla",
                        Collection %in% c("1656", "1654", "1635", "1643", "1645") ~ "East. Ramosa",
                        Collection %in% c("1606", "1560", "1585", "1609", "1611",
                                          "1588", "1626", "1622", "1624") ~ "West. Ramosa"
              )
    )


trait_data <- trait_data %>% 
    mutate(inflor_position = Plant_height/Veg_height,
           relative_sheath_length = Sheath_length/Internode_dist,
           relative_leaf_length = Leaf_length/Sheath_length,
           glume_enclosure = Glume_length_outer/Spikelet_length,
           glume_ratio = Glume_length_outer/Glume_length_inner,
           spikelet_shape = Spikelet_length/Spikelet_width,
           lemma_ratio = Lemma_outer_length/Lemma_inner_length)

trait_data <- trait_data[, c(1:8, 11:22, 25:32)]
trait_data <- trait_data[, -c(4,8,14,15,16,18,19)]


scale_ram <- data.frame(Clade = as.character(trait_data$Clade), scale(trait_data[,c(4:13, 15:21)]))

lda_ramosa <- lda(scale_ram$Clade ~ ., data = scale_ram)

lda.data <- cbind(scale_ram, predict(lda_ramosa)$x)

ggplot(lda.data, aes(LD1, LD2)) +
    geom_point(aes(color = Clade), size = 3) +
    scale_colour_manual(values = my_colours) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))
```


```{r ram_boxplots, fig.height=9, message=F, fig.show='first', fig.cap='Boxplots of morphological traits used in the linear discriminant analysis for the Ramosa clade. See Table 1 for trait decriptions. Where applicable, measurements are in mm.'}
my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank()
    )

a <- univariate_boxplot_trait_data("Clade", "inflor_position") + ylab("Inflorescence position")
b <- univariate_boxplot_trait_data("Clade", "Veg_height")+ ylab("Vegetative height")
c <- univariate_boxplot_trait_data("Clade", "Leaf_width")+ ylab("Leaf width")
d <- univariate_boxplot_trait_data("Clade", "Leaf_length") +ylab("Leaf length")
f <- univariate_boxplot_trait_data("Clade", "Internode_dist")+ ylab("Internode distance")
e <- univariate_boxplot_trait_data("Clade", "relative_sheath_length")+ ylab("Relative sheath length")
ea <- univariate_boxplot_trait_data("Clade", "relative_leaf_length")+ ylab("Relative leaf length")
g <- univariate_boxplot_trait_data("Clade", "Culm_diam")+ ylab("Culm diameter")
h <- univariate_boxplot_trait_data("Clade", "Inflor_length")+ ylab("Inflorescence length")
#q <- univariate_boxplot_trait_data("Clade", "Spikelet_no")
k <- univariate_boxplot_trait_data("Clade", "Pedicel_length_epu")+ ylab("Pedicel length")
n <- univariate_boxplot_trait_data("Clade", "Spikelet_length") + theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1, size = 12, colour = "black"))+ ylab("Spikelet length (mm)")
#l <- univariate_boxplot_trait_data("Clade", "Glume_length_outer")
l <- univariate_boxplot_trait_data("Clade", "glume_enclosure")+ ylab("Glume enclosure")
m <- univariate_boxplot_trait_data("Clade", "glume_ratio")+ ylab("Glume ratio")
#q <- univariate_boxplot_trait_data("Clade", "Glume_length_inner")
#ma <- univariate_boxplot_trait_data("Clade", "Lemma_inner_length")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))
ma <- univariate_boxplot_trait_data("Clade", "spikelet_shape")+ ylab("Spikelet shape")
mc <- univariate_boxplot_trait_data("Clade", "Spikelet_no")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1, size = 12, colour = "black"))+ ylab("Spikelet no.")
mb <- univariate_boxplot_trait_data("Clade", "lemma_ratio")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1, size = 12, colour = "black"))+ ylab("Lemma ratio")
z <- univariate_boxplot_trait_data("Clade", "Striation_no")+ ylab("Striation no.")

#z <- univariate_boxplot_trait_data("Clade", "Spikelet_width") +
  #theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))

null <- ggplot() +
  geom_point(data = means, aes(x = k, y = ce), size = 3.5, alpha = 0) + theme_void()


ggpubr::ggarrange(c,d, null, b,a,e,ea, f,g,h,k,l,m,z,ma,mc, mb,n,
                  ncol = 3, nrow = 6, 
                  align = 'v',
                  heights = c(1,1,1,1,1,1.65))

```



```{r rup_minor_clades, echo=F, warning=F, fig.width=9, fig.height=12, fig.cap="Phylogenetic relationships of the Setacea clade. A) Maximum-likelihood tree constructed using IQTREE with Partitionfinder from 307 concatenated supercontig loci. Red numbers show ultrafast bootstraps (BS). Branch lengths represent number of substitutions. B) ASTRAL tree constructed using IQTREE gene trees from 307 supercontig loci. Branch lengths represent coalescent units (generations/effective population size). Tip branches are an arbitrary length not calculated by ASTRAL III. Red numbers represent local posterior probabilities (LPP). Leafy Tricost. = Leafy Tricostata, Restioid Tricost. = Restioid Tricostata."}

#Rupestris minorclades

if(!require(pacman)){install.packages("pacman", dependencies=TRUE); library(pacman)}
p_load(ggplot2, dplyr, ape, phytools, ggtree, treeio, tidyr)


#ML###


#Rename tips
species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ml$tip.label[tree_ml$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

rup_clade <- extract.clade(tree_ml, 120)

rup_tree <- ggtree(rup_clade, right = T)
rup_dat <- rup_tree$data

rup_dat2 <- rup_dat[!rup_dat$isTip,]
rup_dat2 <- separate(rup_dat2, label, into = c("LRT", "bootstrap"), sep = "/", remove = F)

ml_rup <- ggtree(rup_clade, size = 1, right = T) +
    geom_tiplab(size = 3.7, offset = 0.001) +
    geom_treescale(x = -0.0025, width = 0.005, offset = 0.2, linesize = 0.6) +
    geom_rootedge(rootedge = 0.001) +
    geom_nodelab(data = rup_dat2, aes(label = bootstrap),
                 nudge_x = -0.0021,
                 nudge_y = -0.33,
                 colour = "#DF0101",
                 size = 4) +
    geom_cladelabel(node = 95, 
                    label = "E. Rupestris",
                    colour = "#1D72F5",
                    extend = 0.3,
                    offset = 0.037,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=95, fill="#1D72F5", alpha = 0.15, extend = 0.032) +
    
    geom_cladelabel(node = 89, 
                    label = "Fernkloof",
                    colour = "#DF0101",
                    extend = 0.6,
                    offset = 0.037,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.54,
                    fontsize = 4) +
    geom_cladelabel(node = 94, 
                    label = "",
                    colour = "#DF0101",
                    extend = 0.45,
                    offset = 0.037,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=89, fill="#DF0101", alpha = 0.15, extendto = 0.045) +
    geom_hilight(node=94, fill="#DF0101", alpha = 0.15, extendto = 0.045) +
    
    geom_cladelabel(node = 84, 
                    label = "Uniflora",
                    colour = "#9471B4",
                    extend = 0.3,
                    offset = 0.037,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=84, fill="#9471B4", alpha = 0.15, extend = 0.0355) +
    geom_cladelabel(node = 85, 
                    label = "Restioid Tricost.",
                    colour = "#FF9326",
                    extend = 0.3,
                    offset = 0.037,
                    offset.text = 0.002,
                    angle = -90,
                    align = T,
                    hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=85, fill="#FF9326", alpha = 0.15, extend = 0.037) +
    
    geom_cladelabel(node = 76, 
                    label = "Dodii",
                    colour = "#77CE61",
                    extend = 0.3,
                    offset = 0.037,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=76, fill="#77CE61", alpha = 0.15, extend = 0.033) +
    
    geom_cladelabel(node = 55, 
                    label = "Setacea",
                    colour = "#0089B2",
                    extend = 0.42,
                    offset = 0.037,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
  geom_cladelabel(node = 53, 
                    label = "",
                    colour = "#0089B2",
                    extend = 0.5,
                    offset = 0.037,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4)+
    geom_hilight(node=55, fill="#0089B2", alpha = 0.15, extendto = 0.051) +
    geom_hilight(node=53, fill="#0089B2", alpha = 0.15, extendto = 0.051) +
    
    geom_cladelabel(node = 73, 
                    label = "Scabra",
                    colour = "#F2B950",
                    extend = 0.3,
                    offset = 0.037,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=73, fill="#F2B950", alpha = 0.15, extend = 0.032) +
    
    geom_cladelabel(node = 71, 
                    label = "Leafy Tricost.",
                    colour = "#027368",
                    extend = 0.3,
                    offset = 0.037,
                    offset.text = 0.0012,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=71, fill="#027368", alpha = 0.15, extend = 0.034) +
    
    geom_cladelabel(node = 67, 
                    label = "W. Rupestris",
                    colour = "#CC1577",
                    extend = 0.3,
                    offset = 0.037,
                    offset.text = 0.001,
                    angle = 0,
                    align = T,
                    #hjust = 0.5,
                    fontsize = 4) +
    geom_hilight(node=67, fill="#CC1577", alpha = 0.15, extend = 0.031) +
    geom_cladelabel(node = 89, 
                    label = "blank",
                    colour = "#DF0101",
                    extend = 0.71,
                    offset = 0.049,
                    offset.text = 0.0012,
                    angle = 90,
                    align = T,
                    alpha = 0,
                    #hjust = 0.54,
                    fontsize = 4) +
  theme(plot.margin=margin(0.1, 0.1, 0.1, 0.1))

#Astral ###########



#Rename tips

species_names <- read.csv("/Users/larawootton/Documents/Masters/Draft_dissertation/Data/Sp_local_edited.csv")

species_names$Collection <- as.character(species_names$Collection)
for (i in 1:nrow(species_names)) {
    tree_ast$tip.label[tree_ast$tip.label == species_names$Collection[[i]]] <- paste(
        species_names$Short_name[[i]],
        paste("(",
              species_names$SampleID[[i]],
              ")", sep = ""),
        species_names$Locality[[i]]
        
    )
}

tree_ast$tip.label[tree_ast$tip.label == "uniflora (1572) Cape Peninsula"] <- "uniflora (1572) Cape Pen."

tree_ast$tip.label[tree_ast$tip.label == "tricostata (1594 repeat) Fernkloof"] <- "tricostata (1594 rep.) Fernkloof"

rup_ast_clade <- extract.clade(tree_ast, 137)

rup_tree_ast <- ggtree(rup_ast_clade, right = T)
rup_ast_dat <- rup_tree_ast$data
rup_ast_dat <- rup_ast_dat[!rup_ast_dat$isTip,]
rup_ast_dat[rup_ast_dat == "1"] <- "1.00"
rup_ast_dat[rup_ast_dat == "0.4"] <- "0.40"
rup_ast_dat[rup_ast_dat == "0.6"] <- "0.60"
rup_ast_dat[rup_ast_dat == "0.7"] <- "0.70"


ast_rup <- ggtree(rup_ast_clade, right = T, size = 1, ladderize = T) +
    geom_tiplab(size = 3.7, offset = 0.02) +
    geom_rootedge(rootedge = 0.05) +
    geom_treescale(x = -0.12, width = 0.30, offset = 0.17, linesize = 0.6)+
    geom_nodelab(data = rup_ast_dat, aes(label = label),
                 nudge_x = -0.2,
                 nudge_y = -0.4,
                 colour = "#DF0101",
                 size = 4) +
    geom_cladelabel(node = 50, 
                    label = "W. Rupestris",
                    colour = "#CC1577",
                    extend = 0.24,
                    offset = 1.7,
                    offset.text = 0.05,
                    #angle = 0,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=50, fill="#CC1577",
                 extend = 2.2,
                 alpha = 0.15) +
    
    geom_cladelabel(node = 57, 
                    label = "Uniflora",
                    colour = "#9471B4",
                    extend = 0.5,
                    offset = 2.3,
                    offset.text = 0.11,
                    angle = -90,
                    hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=57, fill="#9471B4", 
                 extend = 2.2, 
                 alpha = 0.15 ) +
    
    geom_cladelabel(node = 58, 
                    label = "Restioid Tricost.",
                    colour = "#FF9326",
                    extend = 0.2,
                    offset = 1.7,
                    offset.text = 0.11,
                    angle = -90,
                    hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=58, fill="#FF9326", alpha = 0.15, extend = 2.3) +
    
    geom_cladelabel(node = 62, 
                    label = "Fernkloof",
                    colour = "#DF0101",
                    extend = 0.2,
                    offset = 1.7,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=62, fill="#DF0101", alpha = 0.15, extend = 2.37) +
    
    geom_cladelabel(node = 70, 
                    label = "E. Rupestris",
                    colour = "#1D72F5",
                    extend = 0.2,
                    offset = 1.7,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=70, fill="#1D72F5", alpha = 0.15, extend = 2.3) +
    
    geom_cladelabel(node = 73, 
                    label = "Scabra",
                    colour = "#F2B950",
                    extend = 0.2,
                    offset = 1.7,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=73, fill="#F2B950", alpha = 0.15, extend = 2.3) +
    
    geom_cladelabel(node = 77, 
                    label = "Leafy Tricost.",
                    colour = "#027368",
                    extend = 0.2,
                    offset = 1.7,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=77, fill="#027368", alpha = 0.15, extend = 2.3) +
    
    geom_cladelabel(node = 81, 
                    label = "Dodii",
                    colour = "#77CE61",
                    extend = 0.2,
                    offset = 1.7,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=81, fill="#77CE61", alpha = 0.15, extend = 2.3) +
    
    geom_cladelabel(node = 86, 
                    label = "Setacea",
                    colour = "#0089B2",
                    extend = 0.2,
                    offset = 1.7,
                    offset.text = 0.05,
                    #angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    align = T) +
    geom_hilight(node=86, fill="#0089B2", alpha = 0.15, extend = 2.9) +
    
    geom_cladelabel(node = 54, 
                    label = "Blank",
                    colour = "red",
                    extend = 0.21,
                    offset = 2.5,
                    offset.text = 0.05,
                    angle = 90,
                    #hjust = 0.5,
                    fontsize = 4,
                    alpha = 0,
                    align = T) +
  theme(plot.margin=margin(0.1, 0.1, 0.1, 0.1))

    
ggpubr::ggarrange(ml_rup, ast_rup, labels = c("A", "B"), font.label = list(size = 20, face = "plain"), label.x = 0.07, label.y = 0.98)



```

```{r rup_gbs, echo=F, warning=FALSE, cache=F, message=F, include = T, fig.height=6.4, fig.cap='Plots of sNMF genomic assignment of individuals within the Setacea clade for A) K = 9, and B) K = 19. Fractions displayed at the top right of the graphs show the number of times a given sNMF repetition recovered the displayed assignments. Bars below the plots identify accession numbers and taxon groups.' }
library(pophelper)
library(gridExtra)
library(label.switching)
library(tidyr)


clump_file <-readQ(here("Data", "calc.clumpp.merged.K9.txt"))

pop_names <- read.table(here("Data", "popmap_rupestris_reordered"), stringsAsFactors=F)

clade_names <- c(
    rep("Setacea", 20),
    rep("Fernkloof", 11),
    rep("Scab", 4),
    rep("Uni", 4),
    rep("Dodii", 12),
    rep("Leafy tri.", 12),
    rep("Restioid tri.", 12),
    rep("Wem", 4),
    rep("Rup.ss", 8),
    rep("East.rup.", 8)
)

group_labs <- data.frame(collection = pop_names, clade = clade_names)
names(group_labs)[1] <- "collection"
group_labs <- separate(group_labs, 
                       col = collection, 
                       into = c("letter", "pop_id"),
                       sep = "B")
group_labs <- group_labs[,2:3]
group_labs$clade <- as.character(group_labs$clade)

my_colours <- c(
    "#60D5FD", #set
    "#027368", #leafy
    "#77CE61", #dodii
    "dark grey",
    "#1D72F5", #E. rup
    "#851D41", #rup
    "#DF0101", #Fernkloof
     #"#F2B950", #scab
    "#9471B4", #uni
    "#FF9326" #restiod
) 


p5 <- plotQ(clump_file, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 2.7,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 4,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 2.1,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 9, 83/100",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            titlehjust = 1,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

#k = 19

clump_file2 <- readQ(here("Data", "calc.clumpp.merged.K19.txt"))

my_colours <- c("#1D72F5","#DF0101","#77CE61", "#FF9326","#A945FF","#0089B2","#FDF060","#FFA6B2","#BFF217","#60D5FD","#CC1577","#F2B950","#7FB21D","#EC496F","#326397","#B26314","#CCEBC5","#A4A4A4","#610B5E")



p2 <- plotQ(clump_file2, 
            showsp = F,
            splab = "",
            barsize = 1,
            grplab = group_labs,
            grplabsize = 2.7,
            grplabheight = 0,
            clustercol = my_colours,
            linepos = 0.9,
            linealpha = 1,
            linesize = 1,
            pointalpha = 1,
            pointsize = 3,
            showdiv = T,
            divtype = 1,
            divgrp = "clade",
            divsize = 1.9,
            grplabpos = 0.35,
            panelratio = c(10,1),
            showtitle = T,
            titlelab = "K = 19, 61/100",
            titlesize = 14,
            titlecol = "black",
            titlespacer = 5,
            titlehjust = 1,
            basesize = 0,
            exportplot = F,
            returnplot = TRUE)

ggpubr::ggarrange(arrangeGrob(p5$plot[[1]]), arrangeGrob(p2$plot[[1]]), nrow = 2,
                  labels = c("A", "B"), font.label = list(size = 19, face = "plain"), 
                  label.x = 0.001, label.y = 0.999)
```

```{r, include = F}

library("vcfR")
library("adegenet")
library("dartR")
library("plotly")
library(tidyverse)

#read.vcf
vcf <- read.vcfR(here("Data", "Final_rupestris_lenient24Apr.vcf"))

#assign pop identifiers
pop.assign<- read.csv(here("Data", "rupestris_pop_assignment.csv"), header=T)
dat.pop.assign<-data.frame(pop.assign$Species_current, pop.assign$Clade)


dat.pop.assign <- dat.pop.assign %>% 
    mutate(clade_labs =
               case_when(pop.assign.Clade %in% c("setacea") ~ "Setacea", 
                         pop.assign.Clade %in% c("rupestris_original") ~ "Western Rupestris",
                         pop.assign.Clade %in% c("dodii_original") ~ "Dodii",
                         pop.assign.Clade %in% c("tricostata_original") ~ "Restioid Tricost.",
                         pop.assign.Clade %in% c("fernkloof_setacea") ~ "Fernkloof",
                         pop.assign.Clade %in% c("uniflora") ~ "Uniflora",
                         pop.assign.Clade %in% c("milner_group") ~ "Leafy Tricost.",
                         pop.assign.Clade %in% c("scabra") ~ "Scabra",
                         pop.assign.Clade %in% c("rupestris_swartberg") ~ "Eastern Rupestris",
                         pop.assign.Clade %in% c("wemmershoek") ~ "Wemmershoek"
               )
    )


#convert vcf to genlight
gl <- vcfR2genlight(vcf)

##Assign pop names so that R knows how to colour the PCoA points
strata(gl)<-data.frame(dat.pop.assign)
setPop(gl)<-~clade_labs

#Before PCoA, find & exclude the SNPs with missing data:
toRemove <- is.na(glMean(gl, alleleAsUnit = FALSE)) 
which(toRemove) 
gl.na <- gl[, !toRemove]
pop(gl.na)

my_colours <- c(
    "#60D5FD", #set
    "#851D41", #rup
    "#77CE61", #dodii
    "#FF9326", #restiod
    "#DF0101", #Fernkloof
    "#9471B4", #uni
    "#027368", #leafy
    "#F2B950", #scab
    "#1D72F5", #E. rup
    "grey"
)  

my_theme <- theme_bw() +
  theme(panel.grid.minor = element_blank(),
        panel.grid = element_blank(),
        legend.key = element_rect(fill = NA, color = "grey"),
        legend.text = element_text(size=12),
        legend.background = element_blank()
  )


pc<-gl.pcoa(gl.na, nfactors=5) 
pcoa_plot1 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=1, yaxis=2)

p1 <- pcoa_plot1 + my_theme +
          scale_colour_manual(values = my_colours, name = " ") +
          geom_point(size = 4)

pcoa_plot2 <- gl.pcoa.plot(pc, gl.na,labels="legend", xaxis=3, yaxis=4)

p2 <- pcoa_plot2 + my_theme +
  scale_colour_manual(values = my_colours, name = " ") +
  geom_point(size = 4) 


```

```{r pcoa_rup, echo=F, warning=FALSE, cache=T, message=F, fig.height=7, fig.width=5, fig.cap='Representation of genetic similarity between individuals in the Setacea clade using a principle coordinates analysis of SNP variants. A) Principal component axes 1 and 2, B) principal component axes 3 and 4.'}

ggpubr::ggarrange(p1, p2, nrow = 2, align = "hv", labels = c("A", "B"), font.label = list(size = 18, face = "plain"), label.x = 0.075, label.y = 0.98,
                  #label.x = 0.12, label.y = 0.98, 
                  common.legend = T, legend = "right")



```

```{r map_rup, fig.width=8, fig.height=7, fig.cap='Map showing average sNMF assignments per population at K = 9 for the Setacea clade. Colours correspond to those of Figure 17A.'}


clump_file9 <- readQ(here("Data", "calc.clumpp.merged.K9.txt"))

gen_dat <- as.data.frame(clump_file9$`calc.clumpp.merged.K9-1`)

pop_names <- read.table(here("Data", "popmap_rupestris_reordered"), stringsAsFactors=F)

gen_dat <- cbind(pop_names, gen_dat)

gen_dat <- separate(gen_dat, 
                    col = V1, 
                    into = c("letter", "Pop"),
                    sep = "B")
gen_dat$Pop[gen_dat$Pop == "LW2"] <- 2


gen_dat$Pop <- as.numeric(gen_dat$Pop)

gen_dat_reduced <- gen_dat %>% 
    group_by(Pop) %>% 
    mutate(C1 = mean(Cluster1),
           C2 = mean(Cluster2),
           C3 = mean(Cluster3),
           C4 = mean(Cluster4),
           C5 = mean(Cluster5),
           C6 = mean(Cluster6),
           C7 = mean(Cluster7),
           C8 = mean(Cluster8),
           C9 = mean(Cluster9)) %>% 
    as.data.frame()

gen_dat_reduced <- gen_dat_reduced[, c(2,12:20)] 
gen_dat_reduced <- gen_dat_reduced %>% distinct()


rup_location <- location_data %>% 
    filter(Major_groups == "rupestris") %>% 
    distinct()


combined <- left_join(gen_dat_reduced, rup_location, by = "Pop")
combined <- combined[,c(1:12, 21:22)]

combined$Edeg[combined$Pop == 1616] <- 18.29265
combined$Sdeg[combined$Pop == 1604] <- -34.45525
combined$Edeg[combined$Pop == 1589] <- 18.78788
combined$Sdeg[combined$Pop == 1594] <- -34.4903
combined$Edeg[combined$Pop == 1600] <- 19.37005
combined$Edeg[combined$Pop == 1597] <- 19.22052
combined$Sdeg[combined$Pop == 1597] <- -34.33863
combined$Sdeg[combined$Pop == 1632] <- -34.09035
combined$Edeg[combined$Pop == 1631] <-  20.3272
combined$Edeg[combined$Pop == 1634] <- 20.57443
combined$Sdeg[combined$Pop == 1633] <- -33.88173
combined$Edeg[combined$Pop == 1633] <- 20.44318
combined$Edeg[combined$Pop == 1559] <- 19.76875
combined$Edeg[combined$Pop == 1565] <- 18.86608
combined$Sdeg[combined$Pop == 1565] <- -34.00277
combined$Sdeg[combined$Pop == 1579] <- -33.98938

combined$Pop[24] <- "LW3"

my_colours <- c(
    "#60D5FD", #set
    "#027368", #leafy
    "#77CE61", #dodii
    "dark grey",
    "#1D72F5", #E. rup
    
    "#851D41", #rup
    "#DF0101", #Fernkloof
    
    #"#F2B950", #scab
    "#9471B4", #uni
    "#FF9326" #restiod
) 

map_res +
    geom_scatterpie(data = combined, 
                    aes(x = Edeg, y = Sdeg, r = rep(0.075, 216)), 
                    cols = colnames(combined[,2:10]),
                    pie_scale = 5,
                    size = 0.05) +
    scale_fill_manual(values = my_colours)+
    geom_text(data = combined, 
              aes(x = Edeg, y = Sdeg),
              label = combined$Pop,
              nudge_y = -0.12,
              size = 2.5
    )+
    theme(legend.position = "none",
          axis.title.y = element_text(colour = "white"),
          axis.title.x = element_text(colour = "white")) +
    coord_equal()

```


```{r lda_rup, fig.cap='A linear discriminant analysis of the morphological data for the Setacea clade. Traits included are described in Table 1. For the full dataset, plot A) shows axes PC1 and PC2, B) shows PC3 and PC4. Restioid tricostata and Uniflora are removed from the analysis in C) and D).'}

my_colours <- c("#77CE61", #dodii
                "#1D72F5", #E. rup
                "#DF0101", #Fernkloof A
                "#FFA5B0", #Fernkloof B FF6F5E
                "#027368", #leafy
                "#FF9326", #restiod
                "#851D41", #rup
                "#F2B950", #scab
                "#60D5FD", #set
                "#9471B4",
                "dark grey") #uni

my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1)
    )

trait_data <- read.csv(here("Data", "trait_dataV3.csv"), sep = ",")
trait_data$Collection <- as.character(trait_data$Collection)

trait_data <- trait_data %>% 
    filter(Species == "E_rupestris_rupestris" | 
               Species == "E_rupestris_dodii" |
               Species == "E_rupestris_tricostata" |
               Species == "E_setacea_setacea" |
               Species == "E_setacea_scabra" |
               Species == "E_setacea_uniflora" |
               Species == "E_setacea_disticha"
    )%>% 
    droplevels() %>% 
    mutate(Clade = 
               case_when(Collection %in% c("1589", "1597", 
                                           "1599", "1593", "1598") ~ "Fernkloof A",
                         Collection %in% c("1600", "1595") ~ "Fernkloof B",
                         Collection %in% c("1614", "1610", "1569", "1568", 
                                           "1587", "1557", "1558", "1629", 
                                           "1633", "1617", "1577", "1564",
                                           "1580", "1582") ~ "Setacea",
                         Collection %in% c("1579", "1661", "1603", "1594") ~ "Restioid Tricost.",
                         Collection %in% c("1639", "1573",
                                           "1578", "1565", "1602", "1616") ~ "Dodii",
                         Collection %in% c("1618", "1619", 
                                           "1621", "1648", "1632", "1586") ~ "Leafy Tricost.",
                         Collection %in% c("1634", "1559", "123",  "1605", "1563") ~ "West. Rupestris",
                         Collection %in% c("1644", "1636", "1631", "1642", "1640", "1641") ~ "Scabra",
                         Collection %in% c("2", "1655") ~ "East. Rupestris",
                         Collection %in% c("1572", "1604") ~"Uniflora",
                         Collection %in% c("1657") ~ "Wemmershoek"
               ))



trait_data <- trait_data %>% 
    mutate(inflor_position = Plant_height/Veg_height,
           relative_sheath_length = Sheath_length/Internode_dist,
           relative_leaf_length = Leaf_length/Sheath_length,
           glume_enclosure = Glume_length_outer/Spikelet_length,
           glume_ratio = Glume_length_outer/Glume_length_inner,
           spikelet_shape = Spikelet_length/Spikelet_width,
           lemma_ratio = Lemma_outer_length/Lemma_inner_length) 
trait_data <- trait_data[, c(1:8, 11:22, 25:32)]
trait_data <- trait_data[, -c(4,8,14,15,16,18,19)]


#LDA

scale_rup <- data.frame(Clade = as.character(trait_data$Clade), scale(trait_data[, c(4:12, 15:21)]))

require(MASS)
lda_rup <- lda(scale_rup$Clade ~ ., data = scale_rup)

lda.data <- cbind(scale_rup, predict(lda_rup)$x)

ld1 <- ggplot(lda.data, aes(LD1, LD2)) +
    geom_point(aes(color = Clade), size = 2) +
    scale_colour_manual(values = my_colours) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))
ld2 <- ggplot(lda.data, aes(LD3, LD4)) +
    geom_point(aes(color = Clade), size = 2) +
    scale_colour_manual(values = my_colours) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))

#remove restiod and uni for clarity

my_colours2 <- c("#77CE61", #dodii
                "#1D72F5", #E. rup
                "#DF0101", #Fernkloof A
                "#FFA5B0", #Fernkloof B
                "#027368", #leafy
                #"#FF9326", #restiod
                "#851D41", #rup
                "#F2B950", #scab
                "#60D5FD", #set
                "dark grey") #uni
scale_rup2 <- scale_rup %>% 
    filter(!Clade == "Restioid Tricost." & !Clade == "Uniflora") %>% droplevels()

lda_rup <- lda(scale_rup2$Clade ~ ., data = scale_rup2)

lda.data <- cbind(scale_rup2, predict(lda_rup)$x)

ld3 <- ggplot(lda.data, aes(LD1, LD2)) +
    geom_point(aes(color = Clade), size = 2) +
    scale_colour_manual(values = my_colours2) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))

ld4 <- ggplot(lda.data, aes(LD3, LD4)) +
    geom_point(aes(color = Clade), size = 2) +
    scale_colour_manual(values = my_colours2) +
    my_theme +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5))

ggpubr::ggarrange(ld1, ld2, ld3, ld4, ncol = 2, nrow = 2, align = "hv", labels = c("A", "B", "C", "D"), font.label = list(size = 18, face = "plain"), label.x = 0.07, label.y = 0.98, common.legend = T, legend = "right")

```

\newpage

```{r rup_boxplots, fig.show='first', fig.width=8, fig.height=10.5, fig.cap='Boxplots of morphological traits used in the linear discriminant analysis for the Setacea clade. See Table 1 for trait decriptions.'}

my_theme <- theme_bw() +
    theme(panel.grid.minor = element_blank(),
          panel.grid = element_blank(),
          legend.key = element_rect(fill = NA, color = "grey"),
          legend.background = element_blank(),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank()
    )


a <- univariate_boxplot_trait_data("Clade", "inflor_position") + ylab("Inflorescence position")
b <- univariate_boxplot_trait_data("Clade", "Veg_height")+ ylab("Vegetative height")
c <- univariate_boxplot_trait_data("Clade", "Leaf_width")+ ylab("Leaf width")
d <- univariate_boxplot_trait_data("Clade", "Leaf_length") +ylab("Leaf length")
f <- univariate_boxplot_trait_data("Clade", "Internode_dist")+ ylab("Internode distance")
e <- univariate_boxplot_trait_data("Clade", "relative_sheath_length")+ ylab("Relative sheath length")
ea <- univariate_boxplot_trait_data("Clade", "relative_leaf_length")+ ylab("Relative leaf length")
g <- univariate_boxplot_trait_data("Clade", "Culm_diam")+ ylab("Culm diameter")
h <- univariate_boxplot_trait_data("Clade", "Inflor_length")+ ylab("Inflorescence length")
#q <- univariate_boxplot_trait_data("Clade", "Spikelet_no")
k <- univariate_boxplot_trait_data("Clade", "Pedicel_length_epu")+ ylab("Pedicel length")
n <- univariate_boxplot_trait_data("Clade", "Spikelet_length") + theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1, size = 10, colour = "black"), axis.ticks.x = element_line())+ ylab("Spikelet length (mm)")
#l <- univariate_boxplot_trait_data("Clade", "Glume_length_outer")
l <- univariate_boxplot_trait_data("Clade", "glume_enclosure")+ ylab("Glume enclosure")
m <- univariate_boxplot_trait_data("Clade", "glume_ratio")+ ylab("Glume ratio")
ma <- univariate_boxplot_trait_data("Clade", "spikelet_shape")+ ylab("Spikelet shape")
mc <- univariate_boxplot_trait_data("Clade", "Spikelet_no")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1, size = 10, colour = "black"), axis.ticks.x = element_line())+ ylab("Spikelet no.")
mb <- univariate_boxplot_trait_data("Clade", "lemma_ratio")+ theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1, size = 10, colour = "black"), axis.ticks.x = element_line())+ ylab("Lemma ratio")
z <- univariate_boxplot_trait_data("Clade", "Striation_no")+ ylab("Striation no.")

#z <- univariate_boxplot_trait_data("Clade", "Spikelet_width") +
  #theme(axis.text.x = element_text(angle = 40, vjust = 0.9, hjust = 1))

null <- ggplot() +
  geom_point(data = means, aes(x = k, y = ce), size = 3.5, alpha = 0) + theme_void()
null2 <- ggplot() +
  geom_point(data = means, aes(x = k, y = ce), size = 3.5, alpha = 0) + theme_void()


ggpubr::ggarrange(b, null, null2, c, d, a,e,ea, f,g,h,k,l,m,ma,mc, mb,n,
                  ncol = 3, nrow = 6, 
                  align = 'v',
                  heights = c(1,1,1,1,1,1.5))

```