---
title: "Chapt 3 Methods"
author: "Lara Wootton"
date: "26/06/2020"
output: pdf_document
bibliography: Chapt2_refs.bib
csl: harvard-university-of-cape-town.csl
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Methods

## Dating analyses

Molecular dating made use of the targeted-enrichment dataset compiled in Chapter 2, but with reduced taxon sampling. Analyses included a single accession per species in the Ramosa, Rehmannii and Setacea clades, as delimited in Chapter 2, and one accession per taxon (species and subspecies) in the Lowlands and Dura clades. In addition, *O. sativa* was included as an outgroup taxon and _Microlaena stipoides_ was included to represent the Australasian clade. The selection of representative accessions was based on sequencing coverage and completeness of morphological data. The possibility of multiple species within the Western Ramosa clade warranted the inclusion of two accessions from this group, while three accessions of _E. calycina_, corresponding to forms ("robust", "gracile" and "Clanwilliam") that likely represent distinct species [@Musker2013], and accessions of all subspecies of both *E. villosa* and *E. brevifolia* were included, resulting in a total of 42 taxa.

Owing to the computational demands of dating methods, running an analysis on the full number of loci was not a viable way forward. Instead, ten sets of 25 supercontig loci were drawn randomly from a pool consisting of the 308 supercontig loci from the ninety percent coverage dataset of Chapter 2, following the methods of @Stubbs2018. In addition, the gene "shopping" SortaDate package [@Smith2018] was used to choose the 25 "best" and 25 "worst" loci, by generating gene trees for each locus with IQTREE v.1.6.11 [@Nguyen2015], and then ranking the gene trees first by their concordance with the ASTRAL species tree, then by the extent to which their evolution is clock-like, and finally their length (-order 3,1,2). The loci within each set were individually aligned using MAFFT v.7.427 [@Katoh2013], with TrimAL v.1.2 being used to remove gappy regions under the -automated1 option [@Capella2009]. Thereafter, the trimmed loci were concatenated and partitioned using AMAS [@Borowiec2016]. IQTREE was used to find which of the substitution models incorporated in BEAST v1.10.4 [@Drummond2007; @Suchard2018] were suitable for each locus with the option -mset JC, HKY, GTR, TN93. BEAUTi v1.10.4 [@Suchard2018] was used to configure xml files for input into BEAST. All loci were parameterised using unlinked substitution models corresponding to the best model found by IQTREE, while clock and tree models were linked across the loci. Taxon relationships were constrained to correspond to the topology of the ASTRAL tree in Chapter 2. Two calibration points with normal priors were used based on the fossil-calibrated tree of @Prasad2011, one constraining the split between _O. sativa_ and _Ehrharta_ at 72.5 Ma (sd = 5.61, 95% CI = [66.89, 78.11]), and the other between _M. stipoides_ and the Setacea clade at 20 Ma (sd = 7.65, 95% CI = [12.35, 27.65]). BEAST analyses were run using a birth-death tree prior and relaxed uncorrelated lognormal clock model, sampling parameters every 1000 generations. Four replicate BEAST runs of 50 million generations per loci set were used to confirm convergence and stationarity through visualisation in TRACER v1.7.1 [@Rambaut2018], and a fifth replicate of 250 million generations was run to ensure that the Estimated Sample Size (ESS) for each parameter had an approximate value of >100. The five runs based on each set of loci were combined using LogCombiner v1.10.4 [@Suchard2018] with the first 10 million generations per replicate excluded as burn-in. An exception was the Sortadate 'best' loci set, which took longer to reach stationarity and required 20 million generations per replicate to be discarded as burn-in. TreeAnnotator v1.10.4 [@Suchard2018] was used to construct maximum clade credibility trees and summary statistics. Analyses were conducted either using CIPRES Science Gateway [@Miller2010] or the facilities of the University of Cape Townâ€™s ICTS High Performance Computing unit.

## Diversification analyses

To test whether rates of diversification differed between clades, BAMM v2.5.0 [@Rabosky2014] was used to model potential rate shifts. Although BAMM can account for incomplete taxon sampling within a tree [@Rabosky2014], this study was explicitly interested in assessing diversification patterns within the Cape _Ehrharta_, and I therefore excluded unsampled _Ehrharta_ species occurring outside the Cape, such as _E. longiglumis_, as well as _O. sativa_ and _M. stipoides_ were excluded from the analysis. Priors for each of the twelve BEAST trees were generated using the function "setBAMMpriors" from the R package _BAMMTools_ [@Rabosky2014b]. For all trees, the expected number of diversification rate shifts was set to one, based on evidence for an adaptive radiation within the Lowlands clade [@Verboom2003; @Verboom2004]. BAMM was run for 3 million generations, sampling every 1000 generations, with run stationarity and ESS being checked using the R package _Coda_ [@Plummer2006]. The 95% credible set of rate shift configurations was extracted with the function "credibleShiftSet", with the marginal posterior-to-prior odds ratio for detecting a rate shift set to five. In addition, the mean rates of diversification within each of the Lowlands, Succulent Karoo, Ramosa and Setacea clades for each BEAST tree were calculated using the function "getCladeRates" [@Rabosky2014b], and significant differences between clades were tested using an Kruskal-Wallis test. To investigate the influence of the postulated new taxa outlined in Chapter 2, the analysis was then repeated with the BEAST trees trimmed to contain only the set of taxa considered by @Verboom2003. For this purpose, only one accession corresponding to each of _E. setacea_, _E. rupestris_, _E. rehmannii_, _E. calycina_, _E. villosa_, and _E. brevifolia_ were retained. The BAMM results were corroborated using MEDUSA [model = "mixed", @Alfaro2009], in the R package _geiger_ [@Pennell2014]. 

## Rates of morphological change

The non-censored model of @OMeara2006 was used in order to evaluate whether rates of morphological evolution in the Lowlands, Ramosa and Setacea clades differed from that in the  the rest of the tree. This method was chosen as it allows for specific, _a priori_ hypothesis testing, rather than searching for rate shifts anywhere on the tree [eg. @Eastman2011]. For each of the ten BEAST trees, the rate of evolution of a single trait (log-transformed to better fit Brownian motion assumptions [@OMeara2006]) within one of the three clades was compared to the evolutionary rate within the rest of the tree using the function "transformPhylo.ML" and the model = "clade" argument from the R package _MOTMOT_ [@Puttick2019]. Sixteen different morphological traits, as described in Chapter 2 and measured on the same individuals as represented in the BEAST trees, were included in the analysis. The variation in selected traits across the tree was visualised using the function "contmap" implemented in the R package _phytools_ [@Revell2012], and a Levene's test for equal variance [@Levene1960], followed by a Tukey HSD post-hoc test, was used to evaluate the extent to which trait variance differed between clades. 

## Rate of non-synonymous to synonymous change



The codeml program from the PAML package [v4.9, @Yang2007] was used to infer synonymous (dS) and non-synonymous (dN) rates of change, and the ratio (dN/dS = $\omega$) between them. . First, the loci contained in ninety percent coverage exon dataset from Chapter 2 for the Cape _Ehrharta_ included in the BEAST analysis were aligned using the codon-aware alignment program MACSE v2.03 [@Ranwez2011]. Summary statistics for each alignment were then calculated using AMAS [@Borowiec2016], and only loci containing <25% gap characters, >20% variable sites, and having no missing taxa were retained for the analysis, leaving a total of 44 loci. The loci were concatenated into a single alignment, and columns containing >5% gaps were removed using TrimAl v.1.2 under the setting -gt 95 [@Capella2009]. As codeml is insensitive to initial branch lengths, a single unrooted BEAST tree was used for all codeml analyses [@Yang2007]. The free-ratios model was implemented to estimate dN, dS and $\omega$ for each branch (model = 1,  NSsites = 0). The dN, dS and $\omega$ of terminal branches were then grouped by clade, and an ANOVA, followed by a Tukey HSD post-hoc test, was used to test for significant differences between clades. In addition, two-rate models were fitted that allowed the $\omega$ of foreground branches, in this case all branches within a given clade, to differ from the remaining background branches [model = 2, NSsites = 0, @Yang1998; @Yang1998b]. A likelihood-ratio test was then used to evaluate whether the two-rate models fitted the data significantly better than a the one-rate branch model (model = 0, NSsites = 0), which estimates a single $\omega$ across the tree.



